<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en-US" xml:lang="en-US">
<head>
  <meta charset="utf-8" />
  <meta name="generator" content="pandoc" />
  <meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
  <meta name="author" content="James Martin" />
  <meta name="author" content="Matthew J. Grainger" />
  <meta name="author" content="Victoria Nolan" />
  <meta name="author" content="William B. Lewis" />
  <meta name="author" content="Erlend B. Nilsen" />
  <meta name="author" content="Sprih Harsh" />
  <meta name="author" content="Chloé R. Nater" />
  <meta name="dcterms.date" content="2023-06-26" />
  <meta name="keywords" content="wildlife, publishing, open science" />
  <title>Chapter 1: Management, Research and Experimental Design</title>
  <style>
    code{white-space: pre-wrap;}
    span.smallcaps{font-variant: small-caps;}
    span.underline{text-decoration: underline;}
    div.column{display: inline-block; vertical-align: top; width: 50%;}
    div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
    ul.task-list{list-style: none;}
    div.csl-bib-body { }
    div.csl-entry {
      clear: both;
    }
    .hanging div.csl-entry {
      margin-left:2em;
      text-indent:-2em;
    }
    div.csl-left-margin {
      min-width:2em;
      float:left;
    }
    div.csl-right-inline {
      margin-left:2em;
      padding-left:1em;
    }
    div.csl-indent {
      margin-left: 2em;
    }
  </style>
  <!--
  Manubot generated metadata rendered from header-includes-template.html.
  Suggest improvements at https://github.com/manubot/manubot/blob/main/manubot/process/header-includes-template.html
  -->
  <meta name="dc.format" content="text/html" />
  <meta name="dc.title" content="Chapter 1: Management, Research and Experimental Design" />
  <meta name="citation_title" content="Chapter 1: Management, Research and Experimental Design" />
  <meta property="og:title" content="Chapter 1: Management, Research and Experimental Design" />
  <meta property="twitter:title" content="Chapter 1: Management, Research and Experimental Design" />
  <meta name="dc.date" content="2023-06-26" />
  <meta name="citation_publication_date" content="2023-06-26" />
  <meta name="dc.language" content="en-US" />
  <meta name="citation_language" content="en-US" />
  <meta name="dc.relation.ispartof" content="Manubot" />
  <meta name="dc.publisher" content="Manubot" />
  <meta name="citation_journal_title" content="Manubot" />
  <meta name="citation_technical_report_institution" content="Manubot" />
  <meta name="citation_author" content="James Martin" />
  <meta name="citation_author_institution" content="Warnell School of Forestry and Natural Resources, University of Georgia, 180 E. Green Street, Athens GA" />
  <meta name="citation_author_orcid" content="0000-0001-8937-2262" />
  <meta name="twitter:creator" content="@gamebirddoc" />
  <meta name="citation_author" content="Matthew J. Grainger" />
  <meta name="citation_author_institution" content="Terrestrial Biodiversity, Norwegian Institute for Nature Research - NINA, Postbox 5685 Torgarden, 7485 Trondheim, Norway" />
  <meta name="citation_author_orcid" content="0000-0001-8426-6495" />
  <meta name="twitter:creator" content="@Ed_pheasant" />
  <meta name="citation_author" content="Victoria Nolan" />
  <meta name="citation_author_institution" content="Warnell School of Forestry and Natural Resources, University of Georgia, 180 E. Green Street, Athens GA" />
  <meta name="citation_author_orcid" content="0000-0002-6069-963X" />
  <meta name="twitter:creator" content="@victoria_nolan1" />
  <meta name="citation_author" content="William B. Lewis" />
  <meta name="citation_author_institution" content="Warnell School of Forestry and Natural Resources, University of Georgia, 180 E. Green Street, Athens GA" />
  <meta name="citation_author_orcid" content="0000-0003-4978-6027" />
  <meta name="twitter:creator" content="@None" />
  <meta name="citation_author" content="Erlend B. Nilsen" />
  <meta name="citation_author_institution" content="Terrestrial Biodiversity, Norwegian Institute for Nature Research - NINA, Postbox 5685 Torgarden, 7485 Trondheim, Norway" />
  <meta name="citation_author_orcid" content="0000-0002-5119-8331" />
  <meta name="twitter:creator" content="@eb_nilsen" />
  <meta name="citation_author" content="Sprih Harsh" />
  <meta name="citation_author_institution" content="Warnell School of Forestry and Natural Resources, University of Georgia, 180 E. Green Street, Athens GA" />
  <meta name="citation_author_orcid" content="0000-0002-6462-7668" />
  <meta name="citation_author" content="Chloé R. Nater" />
  <meta name="citation_author_institution" content="Terrestrial Biodiversity, Norwegian Institute for Nature Research - NINA, Postbox 5685 Torgarden, 7485 Trondheim, Norway" />
  <meta name="citation_author_orcid" content="0000-0002-7975-0108" />
  <meta name="twitter:creator" content="@chloe_nater" />
  <link rel="canonical" href="https://DrMattG.github.io/TechBk1/" />
  <meta property="og:url" content="https://DrMattG.github.io/TechBk1/" />
  <meta property="twitter:url" content="https://DrMattG.github.io/TechBk1/" />
  <meta name="citation_fulltext_html_url" content="https://DrMattG.github.io/TechBk1/" />
  <meta name="citation_pdf_url" content="https://DrMattG.github.io/TechBk1/manuscript.pdf" />
  <link rel="alternate" type="application/pdf" href="https://DrMattG.github.io/TechBk1/manuscript.pdf" />
  <link rel="alternate" type="text/html" href="https://DrMattG.github.io/TechBk1/v/fc76911e75cd64cc6287e00744a174deb12bb5c3/" />
  <meta name="manubot_html_url_versioned" content="https://DrMattG.github.io/TechBk1/v/fc76911e75cd64cc6287e00744a174deb12bb5c3/" />
  <meta name="manubot_pdf_url_versioned" content="https://DrMattG.github.io/TechBk1/v/fc76911e75cd64cc6287e00744a174deb12bb5c3/manuscript.pdf" />
  <meta property="og:type" content="article" />
  <meta property="twitter:card" content="summary_large_image" />
  <link rel="icon" type="image/png" sizes="192x192" href="https://manubot.org/favicon-192x192.png" />
  <link rel="mask-icon" href="https://manubot.org/safari-pinned-tab.svg" color="#ad1457" />
  <meta name="theme-color" content="#ad1457" />
  <!-- end Manubot generated metadata -->
  <!--[if lt IE 9]>
    <script src="//cdnjs.cloudflare.com/ajax/libs/html5shiv/3.7.3/html5shiv-printshiv.min.js"></script>
  <![endif]-->
</head>
<body>
<header id="title-block-header">
<h1 class="title">Chapter 1: Management, Research and Experimental Design</h1>
</header>
<p><small><em>
This manuscript
(<a href="https://DrMattG.github.io/TechBk1/v/fc76911e75cd64cc6287e00744a174deb12bb5c3/">permalink</a>)
was automatically generated
from <a href="https://github.com/DrMattG/TechBk1/tree/fc76911e75cd64cc6287e00744a174deb12bb5c3">DrMattG/TechBk1@fc76911</a>
on June 26, 2023.
</em></small></p>
<h2 id="authors">Authors</h2>
<ul>
<li><p><strong>James Martin</strong>
<br>
<img src="images/orcid.svg" class="inline_icon" width="16" height="16" alt="ORCID icon" />
<a href="https://orcid.org/0000-0001-8937-2262">0000-0001-8937-2262</a>
· <img src="images/github.svg" class="inline_icon" width="16" height="16" alt="GitHub icon" />
<a href="https://github.com/quaildoc">quaildoc</a>
· <img src="images/twitter.svg" class="inline_icon" width="16" height="16" alt="Twitter icon" />
<a href="https://twitter.com/gamebirddoc">gamebirddoc</a><br>
<small>
Warnell School of Forestry and Natural Resources, University of Georgia, 180 E. Green Street, Athens GA
</small></p></li>
<li><p><strong>Matthew J. Grainger</strong>
<br>
<img src="images/orcid.svg" class="inline_icon" width="16" height="16" alt="ORCID icon" />
<a href="https://orcid.org/0000-0001-8426-6495">0000-0001-8426-6495</a>
· <img src="images/github.svg" class="inline_icon" width="16" height="16" alt="GitHub icon" />
<a href="https://github.com/DrMattG">DrMattG</a>
· <img src="images/twitter.svg" class="inline_icon" width="16" height="16" alt="Twitter icon" />
<a href="https://twitter.com/Ed_pheasant">Ed_pheasant</a><br>
<small>
Terrestrial Biodiversity, Norwegian Institute for Nature Research - NINA, Postbox 5685 Torgarden, 7485 Trondheim, Norway
</small></p></li>
<li><p><strong>Victoria Nolan</strong>
<br>
<img src="images/orcid.svg" class="inline_icon" width="16" height="16" alt="ORCID icon" />
<a href="https://orcid.org/0000-0002-6069-963X">0000-0002-6069-963X</a>
· <img src="images/github.svg" class="inline_icon" width="16" height="16" alt="GitHub icon" />
<a href="https://github.com/victorianolan8">victorianolan8</a>
· <img src="images/twitter.svg" class="inline_icon" width="16" height="16" alt="Twitter icon" />
<a href="https://twitter.com/victoria_nolan1">victoria_nolan1</a><br>
<small>
Warnell School of Forestry and Natural Resources, University of Georgia, 180 E. Green Street, Athens GA
</small></p></li>
<li><p><strong>William B. Lewis</strong>
<br>
<img src="images/orcid.svg" class="inline_icon" width="16" height="16" alt="ORCID icon" />
<a href="https://orcid.org/0000-0003-4978-6027">0000-0003-4978-6027</a>
· <img src="images/github.svg" class="inline_icon" width="16" height="16" alt="GitHub icon" />
<a href="https://github.com/wblewis7">wblewis7</a><br>
<small>
Warnell School of Forestry and Natural Resources, University of Georgia, 180 E. Green Street, Athens GA
</small></p></li>
<li><p><strong>Erlend B. Nilsen</strong>
<br>
<img src="images/orcid.svg" class="inline_icon" width="16" height="16" alt="ORCID icon" />
<a href="https://orcid.org/0000-0002-5119-8331">0000-0002-5119-8331</a>
· <img src="images/github.svg" class="inline_icon" width="16" height="16" alt="GitHub icon" />
<a href="https://github.com/ErlendNilsen">ErlendNilsen</a>
· <img src="images/twitter.svg" class="inline_icon" width="16" height="16" alt="Twitter icon" />
<a href="https://twitter.com/eb_nilsen">eb_nilsen</a><br>
<small>
Terrestrial Biodiversity, Norwegian Institute for Nature Research - NINA, Postbox 5685 Torgarden, 7485 Trondheim, Norway
</small></p></li>
<li><p><strong>Sprih Harsh</strong>
<br>
<img src="images/orcid.svg" class="inline_icon" width="16" height="16" alt="ORCID icon" />
<a href="https://orcid.org/0000-0002-6462-7668">0000-0002-6462-7668</a>
· <img src="images/github.svg" class="inline_icon" width="16" height="16" alt="GitHub icon" />
<a href="https://github.com/spharsh">spharsh</a><br>
<small>
Warnell School of Forestry and Natural Resources, University of Georgia, 180 E. Green Street, Athens GA
</small></p></li>
<li><p><strong>Chloé R. Nater</strong>
<br>
<img src="images/orcid.svg" class="inline_icon" width="16" height="16" alt="ORCID icon" />
<a href="https://orcid.org/0000-0002-7975-0108">0000-0002-7975-0108</a>
· <img src="images/github.svg" class="inline_icon" width="16" height="16" alt="GitHub icon" />
<a href="https://github.com/ChloeRN">ChloeRN</a>
· <img src="images/twitter.svg" class="inline_icon" width="16" height="16" alt="Twitter icon" />
<a href="https://twitter.com/chloe_nater">chloe_nater</a><br>
<small>
Terrestrial Biodiversity, Norwegian Institute for Nature Research - NINA, Postbox 5685 Torgarden, 7485 Trondheim, Norway
</small></p></li>
</ul>
<div id="correspondence">
<p>✉ — Correspondence possible via <a href="https://github.com/DrMattG/TechBk1/issues">GitHub Issues</a></p>
</div>
<h2 id="introduction">Introduction</h2>
<!-- Leader: James A. Martin -->
<h3 id="what-is-wildlife-science-and-management">What is wildlife science and management?</h3>
<h3 id="seeking-truth">Seeking truth</h3>
<h3 id="uncertainty">Uncertainty</h3>
<h3 id="environmental-variation">Environmental variation</h3>
<h3 id="linguistic-uncertainty">Linguistic uncertainty</h3>
<h3 id="partial-observability">Partial observability</h3>
<h3 id="partial-controllability">Partial controllability</h3>
<h3 id="structural-uncertainty">Structural uncertainty</h3>
<h2 id="opening-the-gates-of-management-and-science">Opening the Gates of Management and Science</h2>
<!-- Leader: Chloe Nater; co-leaders: Matt Grainger, and Erlend Nilsen -->
<h3 id="openness-to-diversity-of-knowledge">Openness to diversity of knowledge</h3>
<p>Wildlife research and management are closely entwined fields with a large number of different stakeholders. Not only that, but the stakeholders typically represent a highly diverse group of actors with complimentary knowledge and expertise relevant to the wildlife systems and/or management challenges at hand <span class="citation" data-cites="vbxx8YaD">[<a href="#ref-vbxx8YaD" role="doc-biblioref">1</a>]</span>.
This diversity of knowledge can be illustrated through the – admittedly very vague – concept of “expert knowledge”.<br />
In the eyes of the general public it may be the <strong>researchers</strong> that come to mind first when thinking about “experts” [<span class="citation" data-cites="1GZya86fQ">[<a href="#ref-1GZya86fQ" role="doc-biblioref">2</a>]</span>; funk2019]. Researchers collect and analyze data in order to provide a (quantitative) knowledge base for decision-making. Modern science is becoming increasingly more collaborative and typically involves teams of researchers with different and complimentary skills and backgrounds <span class="citation" data-cites="juUi38G9 1HhzKAC1K OdGVIIR2">[<a href="#ref-juUi38G9" role="doc-biblioref">3</a>,<a href="#ref-1HhzKAC1K" role="doc-biblioref">4</a>,<a href="#ref-OdGVIIR2" role="doc-biblioref">5</a>]</span>.
At the same time, there is a growing awareness in the scientific community of the need for interdisciplinary work that also makes use of what is often termed “expert knowledge” in scientific work <span class="citation" data-cites="BQhgQLjo HrWLvwSo">[<a href="#ref-BQhgQLjo" role="doc-biblioref">6</a>,<a href="#ref-HrWLvwSo" role="doc-biblioref">7</a>]</span>.<br />
When used by a researcher, the term “expert” often refers to any non-researcher who possesses valuable knowledge (in the context of a study/problem). In wildlife research, this typically encompasses managers and citizen scientists, and sometimes also includes indigenous people.
<strong>Wildlife managers</strong>, occasionally also referred to as “end users”, “decision makers”, or “policy makers”, are aware of the practical contexts framing wildlife research and often have a good grasp of both logistical and financial implications of different types of monitoring and management strategies and interventions <span class="citation" data-cites="BQhgQLjo">[<a href="#ref-BQhgQLjo" role="doc-biblioref">6</a>]</span>. They may also possess a wealth of (qualitative) knowledge about focal species and ecosystems, as well as about motivations and interests of various other stakeholders in a system <span class="citation" data-cites="vbxx8YaD BQhgQLjo">[<a href="#ref-vbxx8YaD" role="doc-biblioref">1</a>,<a href="#ref-BQhgQLjo" role="doc-biblioref">6</a>]</span>.<br />
Some <strong>citizen scientists</strong> (sometimes also called “community scientists”) may possess similar knowledge, at least about the former, based on experiences from their extensive time spent observing, enjoying, and collecting data on species <span class="citation" data-cites="bF8TZ7uw PI3uanHz">[<a href="#ref-bF8TZ7uw" role="doc-biblioref">8</a>,<a href="#ref-PI3uanHz" role="doc-biblioref">9</a>]</span>. They may also have a good understanding of potential issues and challenges with data collection in the field <span class="citation" data-cites="bF8TZ7uw">[<a href="#ref-bF8TZ7uw" role="doc-biblioref">8</a>]</span>. Last but not least, citizen scientists represent a portion of the general public, and may contribute to debates and decision-making with viewpoints, priorities, and values that are central to a large number of people [<span class="citation" data-cites="PI3uanHz">[<a href="#ref-PI3uanHz" role="doc-biblioref">9</a>]</span>; binley2021].<br />
<strong>Indigenous people</strong> have been living as part of and interacting with the natural environment since long before scientific research as we know it today was conducted [berkes2017]. As a consequence, indigenous communities hold intricate knowledge about their lands and waters which is tightly linked to their culture [jessen2022] and which often covers time spans necessary for understanding long-term ecological changes [e.g. savo2016]. Indigenous knowledge is, in many ways, complementary to ecological science and the involvement of indigenous people can lead to more impactful, effective, and just research and management [ban2018].<br />
Notably, individuals may also belong to several of the categories introduced above. Wildlife managers, for example, may be actively doing research (or have done so in the past) and may also be collecting citizen science data in their free time. Indigenous people may hold central roles in conservation decision making. Some researchers may have indigenous backgrounds, etc. If anything, that only further enhances the diversity of knowledge surrounding wildlife research and management.</p>
<h3 id="engagement-of-diverse-stakeholders">Engagement of diverse stakeholders</h3>
<p>Harnessing and making optimal use of the diversity of available knowledge surrounding wildlife and wildlife management requires broad involvement of different stakeholder groups <span class="citation" data-cites="HrWLvwSo">[e.g. <a href="#ref-HrWLvwSo" role="doc-biblioref">7</a>]</span>. Collaboration is key, and this necessitates that the entire process – from data collection to policy implementation – is inclusive. Four principles form the basis of inclusivity in this context: openness, accessbility, transparency, and reproducibility. These principles are also the corner stones of open science <span class="citation" data-cites="hampton2015 powers2019 vincente2020">[<a href="#ref-hampton2015" role="doc-biblioref"><strong>hampton2015?</strong></a>,<a href="#ref-powers2019" role="doc-biblioref"><strong>powers2019?</strong></a>,<a href="#ref-vincente2020" role="doc-biblioref"><strong>vincente2020?</strong></a>]</span>.
<strong>Open</strong> processes are clearly visible and offer opportunities for anyone to engage with them and contribute. <strong>Accessible</strong> processes are set up and documented in a way that allows everyone to obtain information relevant to them and be able to understand both the information itself and the way it was generated. <strong>Transparent</strong> processes feature complete documentation of every step of the workflow (this includes data collection, analysis, and presentation but also resulting decision-making) and provide all materials that informed and are produced by them. Finally, the mark of <strong>reproducible</strong> processes is that they are provided as complete workflows and based on tools that everyone can access, meaning that anyone (possessing relevant knowledge) can re-run the workflow and arrive at the same conclusions.
Robust decision-making based on diverse knowledge thus requires workflows that are open, accessible, transparent, and reproducible and the next section provides and overview over the components of such workflows (Figure X).</p>
<h3 id="open-and-accessible-workflows">Open and accessible workflows</h3>
<p>When considering research workflows, we often only think of the steps connecting data and results. However, complete workflows in wildife research and management begin with planning and implementation of monitoring and end with reporting, decision making, and evaluation. Knowledge is built incrementally, and workflows should therefore also be thought of as cycles where newly gained information and results from evaluation feed back into earlier steps (e.g. sensu adaptive management, <span class="citation" data-cites="williams2011">[<a href="#ref-williams2011" role="doc-biblioref"><strong>williams2011?</strong></a>]</span>; <span class="citation" data-cites="nichols2015">[<a href="#ref-nichols2015" role="doc-biblioref"><strong>nichols2015?</strong></a>]</span>).</p>
<figure>
<img src="./images/OpenWorkflows_Chart.png" alt="Schematic overview of the components of open and accessible workflows for wildlife research and decision-making." />
<figcaption aria-hidden="true">Schematic overview of the components of open and accessible workflows for wildlife research and decision-making.</figcaption>
</figure>
<p><br/></p>
<p>?? gives an overview over the main components of open and accessible workflows.</p>
<p>It starts with data collection and hence with <strong>monitoring protocols</strong>, which should be openly accessible and documented in a way that they can be understood and – in theory – re-produced/re-implemented by independent parties.<br />
The outcome of monitoring is recorded <strong>raw data</strong>, which should be made available publicly if possible and ideally adhere to FAIR (<em>F</em>indabilty, <em>A</em>ccessibility, <em>I</em>nteroperability, <em>R</em>eusability) and/or CARE (<em>C</em>ollective benefit, <em>A</em>uthority to control, <em>R</em>esponsibility, and <em>E</em>thics) data principles <span class="citation" data-cites="wilkinson2016 carroll2020 carroll2021">[<a href="#ref-wilkinson2016" role="doc-biblioref"><strong>wilkinson2016?</strong></a>,<a href="#ref-carroll2020" role="doc-biblioref"><strong>carroll2020?</strong></a>,<a href="#ref-carroll2021" role="doc-biblioref"><strong>carroll2021?</strong></a>]</span>. Data needs to be supplemented with appropriate and standardized <strong>metadata</strong> to communicate both the structure and content of the data itself, and how it has been generated (i.e. how it is linked to the monitoring protocol).<br />
The process to get from raw data to results typically consists of several steps, e.g. data cleaning, data wrangling/reformatting, data analysis, visualization of results, etc. These steps are implemented using some sort of <strong>code</strong> (manual steps, such as data editing/reformatting in Excel, are best avoided). All code should be well documented, reproducible, and version controlled <span class="citation" data-cites="cooper2017">[see <a href="#ref-cooper2017" role="doc-biblioref"><strong>cooper2017?</strong></a> for a guide]</span> and made available via an appropriate repository such as GitHub, GitLab, etc. Formal code review, as is common in e.g. software development, is also a great tool to enhance reproducibility and overall quality of code <span class="citation" data-cites="ivimey2023">[<a href="#ref-ivimey2023" role="doc-biblioref"><strong>ivimey2023?</strong></a>]</span>.<br />
Under ideal circumstances, both the <strong>software</strong> used for running code (analyses) and the <strong>hardware</strong> on which it is run should be open. For software, this means that the underlying code is open-source and that the program is – ideally – free to use. Defining openness for hardware is trickier, not least because the term spans a large variety of tech ranging from simple field loggers to sophisticated super-computers. These are typically not “publicly available”, but what is crucial in the context of open and transparent workflows is that it is clearly stated what hardware was used and – if applicable – how one may get access to it.<br />
The results from data analysis, their interpretation, and potentially recommendations resulting thereof are presented in <strong>reports</strong>. Whether these take the shape of institutional written reports, scientific articles, oral presentations for different interest groups or any combination of them: they hold important information that should be accessible to anyone interested.<br />
While a majority of written reports nowadays are open access and dissemination presentations often are public events, the same cannot be said of most <strong>decision making</strong> processes. It is not rare for decisions in wildlife management to be made by a small number of individuals behind closed doors. That in itself does not need to be problematic, and may, in many cases, be the most efficient approach to reaching a conclusion and deciding on action. What is important for openness of and trust in wildlife management, however, is that there is an accessible record documenting the decision-making process. That way, anyone interested may gain insights into how and based on what evidence and factors decisions were made.<br />
This is also crucial for the final step of the workflow: <strong>evaluation</strong>. Thoroughly evaluating the different steps of the workflow, from the design of the monitoring to the implementation of management actions and policy, is crucial for improving outcomes and maximizing value for money in the long run.<br />
Finally, it is important to be aware that open scientific and management practices require appropriate <strong>infrastructure</strong>. This includes – but is not limited to – databases and repositories for storing and sharing data, code, and documents. Relevant infrastructure may be institutional or global, as long as it is accessible, endorses standardized formats, and provides both permanent identifiers and version control.</p>
<p>Wildlife research and management need to be open and accessible to harness the diversity of knowledge available, which in term is crucial to tackle current and future challenges. Open and accessible workflows, from planning of monitoring to evaluation of decisions and actions (Figure X), are key to achieving that. Changing practices towards that goal will happen both through adaptation by current researchers and managers, and by teaching the next generation. Successes and failures in setting up and operating with open workflows provide examples that can be used in <strong>education</strong> and help equipping tomorrow’s researchers and managers with the skills they need for successful wildlife management and conservation in a rapidly changing world.</p>
<h2 id="basics-of-managementdecision-science">Basics of Management/Decision Science</h2>
<!-- Leader: James A. Martin; co-leaders: Erlend Nilsen, and Matt Grainger -->
<h3 id="value-of-information">Value of Information</h3>
<h3 id="evidence">Evidence</h3>
<h3 id="proact">PrOACT</h3>
<h3 id="management-strategy-evaluation">Management Strategy Evaluation</h3>
<h3 id="adaptive-resource-management">Adaptive Resource Management</h3>
<h2 id="causation-and-inference">Causation and Inference</h2>
<!--  Leader: Erlend Nilsen; co-leaders: Chloé Nater and Matt Grainger, and James Martin -->
<h3 id="asking-the-right-questions-in-the-right-way">Asking the right questions in the right way</h3>
<h4 id="estimation-questions">Estimation questions</h4>
<h4 id="hypothesis-driven-research">Hypothesis driven research</h4>
<h4 id="exploratory-research">Exploratory research</h4>
<h3 id="causation-and-correlation">Causation and correlation</h3>
<h4 id="sufficient-causation">Sufficient causation</h4>
<h4 id="necessary-causation">Necessary causation</h4>
<h3 id="manipulative-experiments">Manipulative Experiments</h3>
<h3 id="observational-studies">Observational Studies</h3>
<h4 id="directed-acyclic-graphs-dags">Directed Acyclic Graphs (DAGs)</h4>
<h4 id="confounding-variables">Confounding variables</h4>
<h4 id="mediator-and-moderating-variables">Mediator and moderating variables</h4>
<h2 id="basics-of-robust-experimental-design">Basics of Robust Experimental Design</h2>
<!-- Leader: Will Lewis; co-leaders: James Martin -->
<p>Researchers must carefully consider the design of their study before
initiating experiments or collecting observational data. After
formulating a hypothesis, researchers should answer the following
questions: what type of data will need to be collected to assess the
hypothesis, what are the spatial and temporal dimensions of the study,
what treatments (if any) will be used and how will they be applied, how
will study areas or individual animals be selected for study, what
sample sizes will be required, what are the potential sources of error
which could affect the data collection process, how can these potential
sources of error be accounted for, and what are the appropriate
statistical methods for analyzing the data. The process of experimental
design is critical for ensuring that inferences generated from the
collected data will properly assess hypotheses. Inappropriate
statistical methods are relatively easy to rectify if the study design
and data collection are sound; however, there may be little that can be
done statistically to rectify a flawed study design. After designing
your study, solicit review from peers with expert knowledge of your
methods or study system. Consulting a statistician when designing a
study may also be helpful. Soliciting advice and review from experts can
help to improve the design of a study and rectify any overlooked
methodological issues before data collection starts. During this
process, researchers should carefully consider the three cornerstones of
experimental design <span class="citation" data-cites="17087XSCD">[<a href="#ref-17087XSCD" role="doc-biblioref">10</a>]</span>: 1) Randomization, 2) Replication,
and 3) Controls.</p>
<h3 id="randomization">Randomization</h3>
<p>Randomly assigning treatments to units is a fundamental cornerstone of
experimental studies. Indeed, it is the ability of the researcher to
randomly assign treatments that separates experimental studies from
observational studies. Randomly applying treatments to units is
important for preventing conscious or unconscious researcher bias from
affecting the results. Imagine a study testing the hypothesis that
providing corn at feeding stations will increase overwinter survival of
eastern gray squirrels (<em>Sciurus carolinensis</em>). We have 20 study plots
available, 10 of which will be continuously supplemented with dried corn
throughout the winter while the other 10 will not. Some study plots are
far off the road and require a substantial hike through the woods to
reach, while others are situated along the road. Because carrying bags
of corn through the woods is difficult, we decide to supplement corn at
the 10 plots nearest to roads. The problem with this design is that we
have introduced bias by selecting experimental plots nearest to the
roads instead of at random. Squirrels living near the road may naturally
suffer higher rates of mortality from car collisions irrespective of
food abundance. Our data may actually show lower survival in treated
plots and we may erroneously conclude that corn supplementation
decreases survival of squirrels; however, these results are a
consequence of poor study design. A more appropriate design would be to
randomly select plots for food supplementation, regardless of difficulty
of access. Treatments can be randomly selected using a random number
table or random number generator, which are widely available online and
are included in many software packages.</p>
<p>True randomization is not possible for observational studies, as
treatments cannot be applied to units. Surveying abundance of western
capercaillie (<em>Tetrao urogallus</em>) across an elevational gradient would
be an observational study because researchers cannot randomly assign
some locations to be at a high elevation and some to be at a low
elevation. This is not to say that randomization is unimportant for
observational studies; other aspects of randomization can be
incorporated into the design of both observational and experimental
studies to reduce bias. We may not be able to randomly assign the
elevation of potential study locations in the grouse example; however,
if we have a pool of potential survey areas at high and low elevations,
we could randomly select a subset of study locations within each
elevation band.</p>
<h3 id="replication">Replication</h3>
<p>Replication is the process of conducting experimental manipulations or
observations on multiple, independent units. Appropriate levels of
replication must be enacted within all combinations of treatment groups
(or other explanatory variables) relevant to the research hypotheses.
For example, if a researcher is interested in studying the interaction
between food supplementation and predator removal on population growth
rates of northern bobwhites (<em>Colinus virginianus</em>), they would need to
establish multiple study plots for each of the four combinations of food
and predator control treatments (food supplemented/predators removed, no
food supplemented/predators removed, food supplemented/no predators
removed, no food supplemented/no predators removed). Replication is an
essential component of both experimental and observational studies for
several reasons. Replication allows researchers to assess the variance
among units within a treatment group, which is a prerequisite for all
statistical analyses. Moreover, replication reduces the effects of
random chance on the response variable. Researchers are generally not
able to measure every individual in a population, or every potential
study area across the landscape. Instead,researchers generally sample a
subset of the available units and use the measurements from this subset
to estimate the mean of the entire population. Environmental systems are
complex and variable over space, time, and between individuals, so
single observations may represent unusual outcomes rather than the
population mean. Imagine that we are interested in estimating the
juvenile survival and sources of fawn mortality for a mule deer
population (<em>Odocoileus hemionus</em>) and we deploy a radiocollar on a
single fawn. If we found that this fawn was killed by a predator, we
would not conclude that all fawns in this population are killed by
predators. If instead we collared 100 fawns and found that 55 survived,
31 were killed by predators, 6 were killed by vehicle collisions, and 8
died from starvation or disease, the higher number of replicates would
allow us to more confidently conclude that our results are
representative of the actual population values.</p>
<p>Increasing the number of replicates will increase the precision of
population estimates; however, increasing the sample size comes with
increased costs and effort. An important question when designing
experiments is then: what is the sufficient level of replication that
balances precision of estimates with the cost and effort of conducting
the study? The minimum sample size will vary by study, though it may be
estimated by performing a prospective power analysis during the design
phase. Power analyses estimate can estimate the minimum required sample
size for a study based on two parameters: 1) the desired or expected
magnitude of the difference between treatment groups, and 2) the
expected variance between units within treatment group. See <span class="citation" data-cites="steidl2001">[<a href="#ref-steidl2001" role="doc-biblioref"><strong>steidl2001?</strong></a>]</span> for a detailed description of power analyses. The
question, then, is how to estimate the expected variance and difference
between treatment groups before collecting any data. Researchers can use
estimates derived from published literature in a similar study system
with the same or similar species. For example, researchers designing a
study on the effects of predator removal on the survival of northern
bobwhites in Iowa may be able to estimate the minimum number of
radiotransmitters to deploy based on published estimates of the variance
and survival of bobwhites in high-predator and low-predator habitats
from Georgia. If time and funds are sufficient, researchers could also
perform a small-scale test of the methods and data-collection procedures
before the main study. These pilot studies are valuable for power
analyses, as they collect preliminary data on the group means and
variance that are likely to be observed in the main study. In addition
to informing prospective power analyses, pilot studies also allow
researchers to assess the feasibility of the study design and identify
previously-overlooked methodological issues before the main study
commences.</p>
<!-- **Comment: I tried to keep the discussion of the power analyses simple here so I omitted discussing type I and II error rates (since those are set by the researcher in the design phase). I could bring them up, but would require a much more in-depth discussion. Are error rates brought up anywhere else? Also, I could add a section discussing how retrospective power analyses are not really valid, though this would require discussion of power.** -->
<p>Though replication is imperative in scientific studies, researchers must
be careful to avoid pseudoreplication. True experimental replicates are
independent of each other; however, pseudoreplication occurs when
measurements which are not independent are treated as independent in
statistical analyses <span class="citation" data-cites="166u7Adqx">[<a href="#ref-166u7Adqx" role="doc-biblioref">11</a>]</span>. The consequences of
pseudoreplication include overestimating precision and falsely
concluding that there is a treatment effect when there is none <span class="citation" data-cites="16oQ8DJLK">[<a href="#ref-16oQ8DJLK" role="doc-biblioref">12</a>]</span>. Preventing pseudoreplication requires correctly
identifying the experimental unit of a study, which is the lowest
experimental level in which units are independent from others. The
experimental unit is the level at which the treatment is applied in
experimental studies, though care must be taken when identifying the
experimental unit in studies with multiple levels of treatments. Imagine
a study where different herbicides are sprayed on a series of forest
patches, and 4 study plots are established within each patch. Each study
plot is then selectively logged at a different thinning rate. The logged
plots within each forest are not independent, as they are exposed to the
same broad herbicide application. The individual forest patches are the
experimental unit of analysis while the logged plots are considered
subunits. Importantly, increasing the number of subunits will not
increase the levels of replication nor the statistical power of a study;
replication is increased only by increasing the number of experimental
units.</p>
<p>In some cases, non-independence between observations may be readily
apparent. The locations of individual bison (<em>Bison bison</em>) in the same
herd would clearly not be independent of one another, and treating them
as independent would constitute pseudoreplication. In other cases,
pseudoreplication may be less obvious. Imagine that we are designing a
study to assess the effects of selective logging on the body mass of
ruffed grouse (<em>Bonasa umbellus</em>). We identify a forest stand which has
recently been logged and a forest stand which has not been logged, and
in each stand we capture and weigh 30 grouse. We find that the average
mass of the 30 grouse in the logged stand is 377 g and the average mass
of the 30 grouse in the logged stand is 363 g. Doing a t-test with a
sample size of 30 for each treatment, we get a p-value of 0.03 and
conclude that grouse are heavier in selectively-logged stands. The issue
is that, even though we weighed 60 grouse, the treatment (presence or
absence of logging) is applied at the level of the forest stand. Grouse
within a stand are exposed to the same logging application and so are
not independent. The individual forest stands are the experimental units
of analysis while individual grouse are subunits. Even though we have
measured 60 grouse, we only have one forest stand in each treatment. We
have no replication at the level of the experimental unit, and applying
a statistical analysis is inappropriate. We have vastly overestimated
our sample size by treating each of the 60 grouse as independent.
Capturing and weighing more grouse in each patch will not help to
resolve this issue; the only way to increase the number of replicates is
expand our study area to weigh grouse in more logged and unlogged
stands. An appropriate method to avoid pseudoreplication would be to
identify 10 stands which have been recently logged at similar thinning
rates and 10 stands which have not been logged. In each one, we capture
and weigh 30 grouse. We then average the weights across grouse in each
stand to get a single weight value for each experimental unit. We can
then perform a t-test between the average weights of grouse in the 10
logged stands and the 10 unlogged stands.</p>
<p>Though subunits should not be treated as independent replicates,
increasing the number of subunit measurements provides other benefits
for statistical analyses. Sometimes, the variance within experimental
units is of direct interest for studies, e.g., assessing how the
variation in grouse weight rather than mean weight differs between
logged and unlogged stands. In this case, measuring multiple subunits
within each experimental unit would be required to estimate variance.
Taking repeat measurements within experimental units also helps to
reduce stochastic error and more accurately estimate the mean response
in the experimental unit. If we had only captured and weighed one grouse
in each forest stand there is no guarantee that the bird would represent
the population mean; the single grouse may be very light or very heavy.
Measuring 30 birds at each site, however, would likely lead to more
accurate estimates of the stand-level average. The important distinction
to remember is that increasing the number of subunits may increase the
accuracy of measurements within an experimental unit but does not
increase the number of replicates or statistical power.</p>
<p>Replication may be not be feasible in some cases, especially when
assessing hypotheses over broad temporal or spatial scales. Researchers
studying ecosystem-level processes, for example, may often be limited to
a single ecosystem due to the difficulty and cost associated with
manipulating and studying entire ecosystems. In other cases, a study may
be planned to asses the effects of some future development or treatment.
If a dam is scheduled to be built along a river, researchers may assess
the ecological effects of the dam by surveying the fish community before
the dam is constructed and then re-surveying the community after
construction has finished. Similar before/after fish surveying should
also occur in a nearby river which has not been dammed; this will help
assess if changes in the fish community are due to the dam or due to
other sources of variation (i.e., increased pollution from the
surrounding area, introduction of invasive species, etc.). This
experimental protocol is referred to as a BACI design (before-after,
control-impact) and can be a powerful method for assessing the effects
of a single treatment <span class="citation" data-cites="1ElVE5fpk">[<a href="#ref-1ElVE5fpk" role="doc-biblioref">13</a>]</span>; however, it is
inherently unreplicated. There is only one dammed river, and it is
obviously unfeasible to dam more rivers to try to increase the number of
replicates. This lack of replication does not imply that ecosystem-level
or BACI studies are not informative; in fact, unreplicated studies may
be highly valuable and are often the only way to study large, complex,
or difficult-to-manipulate systems [<span class="citation" data-cites="166u7Adqx">[<a href="#ref-166u7Adqx" role="doc-biblioref">11</a>]</span>; doi:10.2307/3802926]. The
important point is that researchers must be careful about generalizing
results from an unreplicated study <span class="citation" data-cites="166u7Adqx">[<a href="#ref-166u7Adqx" role="doc-biblioref">11</a>]</span>. The results from the
BACI study described above may show that the fish community changed
after dam construction, but the results apply only to the effects of
that particular dam. Without replication, there is no way of assessing
if the results from this particular dam are reflective of general
effects of dams on fish communities.</p>
<h3 id="controls">Controls</h3>
<p>The word “control” can apply to multiple different topics in
experimental design <span class="citation" data-cites="17087XSCD">[<a href="#ref-17087XSCD" role="doc-biblioref">10</a>]</span>. One of the most common uses is to
refer to a control treatment, which is an experimental unit that has
either received no treatment or a sham treatment. Proper control
treatments are a critical component of experiments, as they provide a
baseline value with which to compare results from experimental
treatments. If we are interested in determining how vegetation cover
affects nest success of common pheasants (<em>Phasianus colchicus</em>), we
could locate 30 nests, remove 50% of the vegetation within 2 m of the
nest, and monitor nest success. We may find that the success rate of
these manipulated nests is 7% (95% confidence intervals ranging from 4 –
24%), but this value is uninformative on its own. Maybe pheasants
normally only fledge 7% of their nests, and our manipulations had little
effect. In order to discriminate the treatment effect, we need a control
treatment to determine the baseline nest success in this population. If
we monitor an additional 40 unmanipulated nests and find that the nest
success of these control nests is 40% (95% confidence intervals ranging
from 26 – 56%), we can determine that reducing vegetation cover
decreased nest success. Observational studies cannot have true control
treatments; instead, inferences are drawn by assessing how the response
variable is affected by variation in the explanatory factors. For
example, a resource selection study may compare the vegetation
composition and structure between locations used by radio-collared deer
and locations in nearby unused areas. Similarly, a researcher could
estimate the survival of adult northern bobwhites in patches which vary
in hawk density to assess the effects of hawk predation on bobwhite
survival.</p>
<p>Proper inferences depend on control and experimental treatments
differing ONLY in the experimental variable of interest. An
unmanipulated treatment may therefore not serve as an appropriate
control. Suppose we are concerned that the the added weight from
deploying radiotransmitters on breeding female black-throated blue
warblers (<em>Setophaga caerulescens</em>) increases stress and causes them to
abandon their nests. The most reliable way to catch and deploy
transmitters on breeding females of this species is to locate their
nests, place a mist-net nearby, and flush them off their nest into the
net. We perform a study locating 50 nests and randomly assign 25 to the
experimental treatment. We then go out to these experimental nests, set
up a net, flush the female, and attach a transmitter to her. For the 25
control nests, we continue monitoring the nests but otherwise do not
manipulate them. We find that females abandonment occurred at 6 of the
25 experimental nests but only 1 of the 25 control nests, so we conclude
that transmitter deployment increases rates of nest abandonment. This
conclusion is problematic because our control treatment differed from
the experimental treatment in more than one way: experimental females
were fitted with transmitters AND were flushed off their nests,
captured, and handled by the researchers. The increased nest abandonment
rates could have been due to attaching transmitters; however, the
increased rate of nest abandonment could also be due to the disturbance
around the nest. There is no way to separate these possible effects
under the current study design. A more appropriate control treatment,
capturing and handling females at all nests but only deploying
transmitters at experimental nests, would allow for discrimination
between these two potential mechanisms. Similarly, an appropriate
control for a food-supplementation study would be to establish and
periodically visit feeding trays in both experimental and control plots,
but only supply food at the experimental trays. This type of control
would allow researchers to discriminate between the actual treatment
effect and any inadvertent effects related to carrying out the
experiment (e.g., human disturbance, neophobia, etc.). When designing a
control treatment, researchers should endeavor to keep the experimental
methods as similar as possible between experimental and control units so
that only the factor of interest varies.</p>
<p>The word “control” can also refer to mitigating the effects of
extraneous variation on the study results. Extraneous variables are
those that affect the response variable but are not the main variable or
variables of interest. Imagine that we wish to assess insecticide
spraying affects the body condition of a forest-dwelling insectivorous
songbird: the red-eyed vireo (<em>Vireo olivaceus</em>). Our study takes place
in a heterogeneous landscape with forest patches surrounded by
non-forest (Figure 1.1) . We identify 28 study plots scattered among the
8 forest patches, 14 of which we randomly assign to the experimental
group and 14 of which we randomly assign to the control group. We fly a
helicopter over each study plot, spraying insecticides over experimental
plots and water over control plots. We capture and weigh vireos in every
study plot two weeks after spraying. WE then calculate the average vireo
mass for each plot. This study design incorporates randomization,
replication, and an appropriate control treatment; however, many
environmental variables may affect the body mass of vireos besides just
the insecticide treatment. Maybe some forest patches are lower-quality
habitat for vireos than are others (naturally lower food levels, higher
predator density, sub-optimal vegetation, etc.). In this case, vireos in
poor-quality habitats might be expected to have lower body mass than
birds in high-quality habitats regardless of the insecticide treatment.
If we do not account for this confounding variable, then we may find
that the majority of control or treatment plots are located in
poor-quality habitats just by random chance (Figure 1.1 A). This would
lead us to erroneously conclude that insecticide application does not
affect vireo mass (Box 1.1 A) simply because the effect of the
insecticide treatment is conflated with the effect of habitat quality.
Researchers must always consider potential sources of extraneous
variation and how these can be controlled for when designing studies.</p>
<p>Sources of extraneous variation can be controlled for statistically or
through modifications of the study design. In the vireo example, the
extraneous effect of habitat quality could be removed from analysis by
modifying the study design to only sample in either high-quality or
low-quality habitats (Figure 1.1 B, Box 1.1 B). This method of
restricted sampling may be impractical in many cases, especially if the
number of possible sampling locations is small relative to the number of
confounding environmental gradients. Extraneous variables may also be
controlled through a process called blocking, wherein experimental units
are grouped into relatively homogeneous units. If we think that patches
differ in habitat quality, and these differences may introduce
extraneous variation into our analysis, we can treat each forest patch
as a “block” and distribute control and experimental treatments randomly
within each block (Figure 1.1 C). Distributing control and experimental
treatments evenly across blocks ensures that the effect of the block on
the response variable is evenly distributed across treatments. Moreover,
we can account for the variation introduced by the block during
statistical analyses. Including habitat quality as a covariate in the
statistical model allows us to separately estimate the effects of
insecticide application and habitat quality on vireo mass (Box 1.1 C),
thereby leading us to correctly conclude that insecticide use has an
effect on vireo body mass.</p>
<p><img src="BlockingFigure.png" alt="Figure 1.1. Example study designs distributing experimental (E) and control (C) study plots across patches varying in habitat quality. High-quality patches are green while low-quality plots are blue. Study designs differ in how they control for extraneous envrionmental variation: not accounting for envrionmental variation (A), accounting for environmental variation by restricting the effects to a single level of habitat quality (B), or accounting for environmental variation through blocking across levels of habitat quality (C)." />
<img src="https://user-images.githubusercontent.com/10168347/205140795-c086b739-3f27-41cc-a76a-5ee9aa4a3835.png" alt="BlockingFigure" /></p>
<p><img src="blocking.png" alt="Box 1.1. Example R-code showing the effects of controlling variation through restricting sampling and blocking. In this example, we wish to assess the effects of insecticides on the body mass of red-eyed vireos (Vireo olivaceus). Study plots are distributed across 8 different forest patches which vary in habitat quality." />
<img src="https://user-images.githubusercontent.com/10168347/205140839-3a1c644e-d6b7-4203-83b3-9522afb4d816.png" alt="blocking" /></p>
<p>Blocking is not just valuable for experimental studies, but should be
carefully considered when designing observational studies as well.
Imagine a study surveying clearcut and mature forest patches in
mountainous areas to determine how the abundance of rabbits is affected
by logging. We may reasonably expect that rabbit abundance will vary
with elevation in this study. We could incorporate blocking to control
for this extraneous variation during the design phase of the study; for
example, by pairing each clearcut patch with a nearby mature forest
patch of similar elevation. Implementing a blocked study design is a
powerful method of accounting for known or hypothesized confounding
variables in both experimental and observational studies, but can also
help guard against extraneous variation from unforeseen environmental
variation <span class="citation" data-cites="166u7Adqx">[<a href="#ref-166u7Adqx" role="doc-biblioref">11</a>]</span>.</p>
<p>Not all sources of extraneous variation can be easily controlled for
through modifications of the study design, but many of these may be
controlled for during statistical analyses. If we are studying the
effects of age on late-winter survival of mourning doves (<em>Zenaida
macroura</em>), we may capture doves in the summer, age them as either
juveniles or adults based on plumage characteristics, deploy
radiotransmitters, and track them throughout the late-summer to
determine survival rates. Survival is likely to dependent on multiple
environmental factors besides age; for example, survival may also be
positively correlated with body size and local food abundance. We could
attempt to control for these sources of extraneous variation by blocking
doves into different levels of food abundance and size classes, though
it may be difficult and costly to catch and track enough juvenile and
adult doves in each food/size class to provide adequate replication.
Food availability and size are also continuous variables, so it is more
appropriate to treat them as such in the analyses. In contrast to
blocking designs, these continuous extraneous variables can easily be
controlled for in statistical models (Box 1.2). Similarly, the
probability of detecting individuals during a survey is often dependent
on environmental conditions. When performing avian point counts, for
example, the probability of birds singing and being detected by
observers can depend on factors such as the time of day, precipitation,
temperature, wind speed, and background noise. Careful experimental
design may reduce some of this extraneous variation (only doing surveys
in the morning, not doing surveys in heavy rain or wind); however,
blocking or performing all surveys under the exact same environmental
conditions is clearly infeasible. Statistical control during analyses is
needed to account for these environmental variables. Researchers may not
always known <em>a priori</em> all environmental gradients affecting the
response variable; therefore, it is good practice when collecting data
to record as much information about environmental conditions as
possible. It is generally better to collect extra data that ultimately
is not used then it is to collect insufficient data and end up regret it
later.</p>
<p><img src="statisticalcontrol.png" alt="Box 1.2. Example R-code for controlling extraneous variation during statistical analyses. In this example, age-related differences in late-summer of survival of mourning doves (Zenaida macroura) are of interest; however, body size (as measured by wing length) and food abundance also affect survival." />
<img src="https://user-images.githubusercontent.com/10168347/205140978-8162fd32-0ddd-4306-91db-8150ad0bdbaa.png" alt="statisticalcontrol" /></p>
<h2 id="sampling">Sampling</h2>
<p>The ecological question often revolves around measuring relevant biological variables (survival, density, etc.) on items (like populations, species, subspecies, etc.) of interest in the system under study. In these questions, researchers conduct studies to obtain accurate estimates of characteristics that are important for management and conservation decisions. A vast amount of literature on the subject of <strong>sampling or finite population sampling</strong> (Cochran 1963) exists to help researchers in planning these studies.</p>
<p>The set of all possible individuals from which processes and patterns are to be deduced is called the <strong>target population</strong>. Rarely in any ecological studies, a census of the population is possible. Even in the case of rare and critically endangered species (less in number) or the case of immobile organisms like plants, collecting data from the whole population can be impossible. Thus, in most cases, researchers study a <strong>subset of the population (sample)</strong> and use collected information to draw inferences about the target population. The sample therefore can be described as a group of individuals who participate in research and represent the whole population. In more scientific terms, a sample is a subset of a population randomly selected based on some probabilistic design. In ecological studies, we often hear the term <strong>sampling frame (or study area)</strong>. To efficiently design these studies, it is necessary to understand what this term means. It is a finite set of all individuals that could be measured, and we can use different sampling schemes to obtain items from this frame. The sampling frame usually coincides with the target population, but reasons like accessibility, logistics, budget, etc. can make it differ otherwise. From this sampling frame, we draw a subset or sample of individuals, the <strong>sampling units</strong>, and the items to be measured for different biological variables. Sampling units should be distinct and easy to define (Box 1, Fig. 1). Target population does not always mean the number of animals. It can also be described in terms of geographic area, in which case, the sampling unit would be grids or township or county depending on the research question.</p>
<p>Sampling is a critical part of both descriptive and experimental studies. All field studies require appropriate sampling designs to reduce variation among observations in the study. The choice of sampling method will depend on the objectives of the study, the distribution, and characteristics of the population being sampled among many other factors.</p>
<p><img src="https://user-images.githubusercontent.com/18090572/199079676-1affc4d3-21fe-4b85-b68c-d3c224f05887.PNG" alt="box1" /> <img src="https://user-images.githubusercontent.com/18090572/199079818-9a89a16a-3442-4ac4-8778-7273c1296228.png" alt="Fig1" /></p>
<p><strong>Fig. 1</strong> A target population of greater sage-grouse (<em>Centrocercus urophasianus</em>, hereafter grouse) in one county in Wyoming, USA. For reasons like accessibility and permission, the whole county area could not be used for conducting a study. So, the sampling frame represents the area from which sampling units could be collected.</p>
<h3 id="probability-and-non-probability-sampling-methods">Probability and non-probability sampling methods</h3>
<p>For a sample to correctly represent a population, we need to properly identify the target population, followed by identifying the sampling frame, sampling units, and sampling technique. We should also be cautious about resource availability-manpower, logistics, time, etc. So, the next question here is how to draw a sample. There are mainly two types of sampling methods: probability and non-probability sampling. These designs differ in terms of the quality of parameter estimates (Box 2).</p>
<p><strong>Probability sampling</strong>: In the probability sampling scheme, every unit from the sampling frame has a non-zero probability of selection. It, therefore, leads to unbiased estimates of the mean and variance for the variable of interest. Therefore, any method aimed at generalizing results drawn by a sample to the whole population of interest must be based on probability sampling.</p>
<p><strong>Non-probability sampling</strong>: In non-probability sampling, researchers select samples based on their convenience. In other words, the researchers purposively choose particular units for constituting a sample. Social researchers often use this sampling design to select households or families to conduct their surveys. A few common techniques are <strong>convenience sampling</strong> and <strong>judgment sampling</strong>. In convenience sampling, samples are chosen based on an arbitrary selection procedure. It is often justified based on accessibility and availability of resources like time, budget, etc. In wildlife studies, this sampling scheme is often used for conducting roadside bird surveys, surveys for identifying mammal tracks near roads, etc. Since the location of the target species decides the sampling frame and number of samples, the results from these studies are often highly biased and far from accurate. In judgment sampling, sampling frame, and samples are chosen based on expert knowledge of the system. One common example from wildlife studies is selecting and classifying <strong>study area (sampling frame)</strong> into low-quality and high-quality based on expert knowledge about the area.</p>
<p>There are pros and cons for each of these sampling designs. Probability sampling helps in reducing sample bias and therefore provides an accurate representation of the population. Non-probability sampling is useful when we still need some preliminary data within time and budget constraints.</p>
<figure>
<img src="https://user-images.githubusercontent.com/18090572/202543160-148904d0-f3cc-4a45-a0e3-ac71ec5bbbe6.PNG" alt="box2a" />
<figcaption aria-hidden="true">box2a</figcaption>
</figure>
<figure>
<img src="https://user-images.githubusercontent.com/18090572/202543180-bf7a2e56-e8b4-401f-9281-6147ac4f399b.PNG" alt="box2acont" />
<figcaption aria-hidden="true">box2acont</figcaption>
</figure>
<h3 id="sampling-designs-under-the-probability-sampling-method">Sampling designs under the probability sampling method</h3>
<p>Although simple random sampling is the most basic technique for sample selection, there are others that are often used in wildlife ecology studies. There are pros and cons associated with each of these designs and they can also be combined to provide a larger set of options for study designs.</p>
<p><strong>Simple random</strong>: Simple random sampling is the process of selecting n units from a population of N units such that every unit has an equal probability of inclusion in the sample. Simple random sampling requires that each sample unit be selected independently of all other units. This method should be used if the area of interest is homogeneous with respect to the elements and characteristics of interest. A simple random sample may be obtained by following basic steps: 1) the study area must be completely covered by non-overlapping sampling units, 2) the population of sampling units is assumed to be finite, 3) sampling units can be located and the measurement of the characteristic of interest on the unit is possible. Also, the error in measuring should be small compared to the differences in the attribute from unit to unit, and 4) sample units are sampled without replacement. For example, suppose a farmer is interested in evaluating the health of his cattle, and each of them is ear tagged. So, under simple random sampling, he can generate some random number using a calculator or excel sheet and select those tagged cattle for health assessment.</p>
<p><strong>Systematic</strong>: Systematic sampling is the process where sampling units are selected at regular intervals. Under this sampling technique, the sampling frame will be partitioned into n number of primary units and then the selection of units will occur in a systematic fashion based on a random start. This design is easier to execute than simple random sampling. In the above example of assessing cattle health, systematic sampling will be easier to implement in case the cattle are not ear-tagged. Then under systematic design, the farmer can choose every nth cow while they are entering the barn (assuming they follow a queue). Systematic sampling is commonly used to sample vegetation characteristics. For example, determining vegetation characteristics every 10 meters along a line transect in a plot is a classical example of systematic sampling. Systematic sampling has also been criticized in cases when the arrangement of units may follow some pattern in the response variable. For example, let’s say we were interested in the number of people using public areas for birding. We decided to establish a check station and take a count using a systematic sample of days during the study period. This could give us a biased result if every sampled day fell on a work week, then the estimates obtained would be very different from estimates obtained from the weekend count.</p>
<p><strong>Stratified</strong>: In wildlife studies, populations tend to be aggregated or clustered, thus sample units closer to each other will be more likely to be similar. For this reason, systematic sampling tends to overestimate the variance of parameter estimates. A uniform grid of points or parallel lines may not encounter rare units. To increase the likelihood of capturing some of these rare units, scientists may stratify the sample such that all units of each distinct type are joined together into strata and simple random samples are drawn from each stratum. Stratified sampling is, therefore, generally used when the population from which the sample is to be drawn does not belong to one homogeneous group and there is a high variation within the population. If, however, the population belongs to a heterogeneous group, the estimates based on earlier sampling designs will be imprecise. If we have prior information associated with the heterogeneity in the population, we can use designs like stratified sampling to select samples which will increase estimates precision. This sampling technique divides the whole population into different mutually exclusive groups (strata) according to some characteristics such as the habitat they inhabit, gender, etc. Ideally, the strata should be homogeneous with respect to the variable of interest (like density, abundance, etc.). This process requires more effort than random sampling but is generally more accurate in terms of representing the population. There is also a limit to the number of strata into which a population can be subdivided. The stratified sampling method is common in wildlife studies, as it helps estimate and contrast parameters among strata. The formal procedure of stratified sampling follows a few steps: 1) specify strata, which must be mutually exclusive, 2) classify all sampling units into their stratum, and 3) draw a simple random sample from each stratum.</p>
<p><strong>Cluster</strong>: A probabilistic sampling scheme in which each sampling unit is a cluster of items such as the group of animals. Cluster sampling is generally used in cases when there are predefined groups within the population. These groups can be based on demographics, habitats, geography, etc. The sampling process starts by dividing the population into small groups known as clusters followed by random selection of these clusters to create a sample. This approach has wide applications in wildlife study as many birds and mammals occur in groups during or all parts of the year. Cluster sampling is useful when the cost or time to travel from one sample unit to the next is too high. Cluster sampling can also be performed in stages. Single-stage cluster sampling happens when all the elements of the chosen clusters are included in the sample. Two-stage cluster sampling is when in contrast to single-stage cluster sampling only some units are observed. The formal procedure of cluster sampling follows a few steps: 1) specify appropriate clusters and make a list of all clusters, 2) draw a simple random sample of clusters, and 3) measure all elements of interest in each selected cluster.</p>
<p><strong>Adaptive sampling</strong>: In various studies, numerous sampling designs are combined under an adaptive sampling framework. In this technique, we start with an initial probabilistic sample of units and add more units in some pre-defined neighborhood or pre-defined condition to this sample (Thompson and Seber 1996, Williams et al. 2002, Thompson 2003). This process continues until no sampled units satisfy the specified condition. Adaptive sampling offers biologists a way to augment the probability sample with samples from other units without losing the benefits of the original probabilistic design. Rules for the selection of additional samples are established based on some characteristic of the variable of interest (e.g., presence/ absence, age, sex, and height).</p>
<p><strong>Case study</strong>: Scientists are worried that ongoing human-induced landscape changes have threatened a grouse population in eastern Wyoming, USA. They decided to conduct a study with two main objectives. They primarily want to understand the impact of this dynamic landscape on the survival of grouse and on lek numbers. The study will be done in Carbon County for a period of 3 years during which they plan to capture 150 grouse in total (both male and female; Fig 5a). For carrying out this study, they divided the study area into 25 equal-sized grids (Fig. 5b). To estimate the survival of grouse, we need to capture individuals and track them for the required time period or till they are alive, whichever comes first. We need to devise a sampling mechanism to select grids from where these grouse can be captured.</p>
<figure>
<img src="https://user-images.githubusercontent.com/18090572/199080003-31925669-0d74-4497-a39f-4303b51b29ed.PNG" alt="Fig5ab" />
<figcaption aria-hidden="true">Fig5ab</figcaption>
</figure>
<p><strong>Fig. 5</strong> a) Carbon County in Wyoming, USA for conducting a 3-year study on a grouse population in the sampling frame, b) Study area divided into 25 equal-sized grids for carrying out the study.</p>
<p>In this case study, we will show you how researchers can sample grids using different types of probabilistic sampling schemes. Under random sampling, each of the grids will have an equal probability of getting sampled (Fig. 6a). In a systematic sampling framework, researchers can pick grids at regular intervals. So let’s say, they decided to select every 5th grid starting with the 3rd grid. So, 3, 8, 13, 18, and 23 will be their plots from which the grouse will be then captured (Fig. 6b).</p>
<p>Suppose researchers identified three different dominant land cover types (grassland, wetland, and forest), so the random selection of grids can follow a stratified framework. Under this framework, researchers will divide the whole study area into these different strata and then randomly select grids from each of these strata to capture grouse (Fig. 6c).</p>
<p>Lek sites are known to be spatially close to each other. So, we would expect that if a lek is inhabiting one primary sample plot (grid), there are other leks in the neighborhood. So here researchers can use cluster sampling to calculate lek numbers. Under single-stage cluster sampling, they would start by randomly selecting primary sample plots across the study area, then within those primary plots, conduct surveys to calculate all lek numbers within the cluster of all four secondary plots (Fig. 6d). In our example, grids can act as primary plots. In each primary sample plot, there are 4 secondary sample plots (numbered 1 to 4 in some order), so under two-stage cluster sampling, we will randomly select any 2 from these 4 secondary sample plots (Fig. 6e). We will then conduct surveys to calculate lek numbers in 2 of these secondary sample plots.</p>
<p>Under adaptive cluster sampling, we will start with the grids which have the highest number of leks and then will sample the next grids with a similar number of grouse (Fig. 6f). See Box 3 for R codes for each of these sampling designs.</p>
<figure>
<img src="https://user-images.githubusercontent.com/18090572/199080044-a08b15a7-e2d8-4034-8cf2-77919c05e49d.PNG" alt="all" />
<figcaption aria-hidden="true">all</figcaption>
</figure>
<p><strong>Fig. 6</strong> Examples of sampling design a) simple random, b) systematic, c) stratify, d) single-stage, e) two-stage, and f) adaptive (numbers in brackets shows the number of leks in each grid) for selecting grids from an equal-sized gridded system in Carbon County, USA for studying grouse population.</p>
<figure>
<img src="https://user-images.githubusercontent.com/18090572/199080135-99771e65-1295-4db1-9d0f-f220138de659.PNG" alt="codes1" />
<figcaption aria-hidden="true">codes1</figcaption>
</figure>
<h3 id="sampling-methodology">Sampling methodology</h3>
<p>In wildlife studies, there are a few commonly used sampling methodologies.</p>
<p><strong>Plots</strong>: Plots are widely used to study habitat characteristics, vegetation characteristics, counting animal numbers, etc. Plots’ shape can vary from circular to square and represents a geographically defined target population. Wildlife tends to be distributed nonrandomly across the landscape in correspondence to the distribution of their habitat. Their distributions are further impacted by intraspecific and interspecific interactions. Given that distributions and abundance vary, plots should vary in shape and size depending on the studied species. Numerous factors influence plot size, including the biology of the species, their spatial distribution, study objectives, logistical considerations, and cost constraints. For example, larger species with large home ranges require larger plots to include adequate numbers. A 3,500-ha plot might include only 10% of the home range of a grizzly bear (<em>Ursus arctos horribilis</em>), the same area could include the entire home ranges of multiple white-footed mice (<em>Peromyscus leucopus</em>). Krebs (1999) listed three main approaches to determining optimal plot shape and size for a study: 1) plot size should have the highest precision for a specific study area, 3) plot size which is most accurate and efficient to answer the question of interest, and 3) plot size which is logistically easy to construct and use.</p>
<p><strong>Points</strong>: In point sampling, a set of points is established throughout the population, and measurements are taken from each point. A common example is a point-count survey for birds where the distance to each heard or seen bird species of interest is measured from a particular point. Selection of sample points can follow any sampling design, as long as points are spaced apart enough that overlapping of the population between points is a bare minimum.</p>
<p><strong>Transects</strong>: In line transects a line or series of lines is randomly or systematically located in the study area. Objects are recorded on either side of the line according to some rule of inclusion. The observer traverses each line, recording the perpendicular distance from the line to each detected animal. These distances are used to estimate the effective width of the area sampled by the transect. Transects can be established using any sampling design as long as each of them is treated as an independent observation and are non-overlapping.</p>
<h3 id="references">References</h3>
<p>Cochran, W. G. 1963. Sampling Techniques, 2nd Ed., New York: John Wiley and Sons, Inc.</p>
<p>Krebs, C. J. 1999. Ecological methodology. Second edition. Benjamin Cummins, Menlo Park, California, USA.</p>
<p>Thompson, S. K. 2003. Editorial: Special issue on adaptive sampling. Environmental and Ecological Statistics 10:507</p>
<p>Thompson, S. K., and G. A. F. Seber. 1994. Detectability in conventional and adaptive sampling. Biometrics 50:713-724</p>
<p>Williams, B. K., J. D. Nichols, and M. J. Conroy. 2002. Analysis and management of animal populations. Academic, San Diego, California, USA</p>
<p>Zar, J. H. 1999. Biostatistical analysis. 4th edition. Prentice/ Hall, Upper Saddle River, New Jersey.</p>
<h2 id="the-publication-and-peer-review-process">The Publication and Peer Review Process</h2>
<!-- Leader: Victoria Nolan; co-leaders: Sprih Harsh -->
<h3 id="introduction-to-peer-review-and-publication">Introduction to Peer Review and Publication</h3>
<p>Effective dissemination and communication of scientific aims, methodology and results is a crucial component of research. Publishing research findings facilitates idea development and sharing within the scientific community and to a wider wildlife audience, including landowners, conservationists and key decision makers keen to implement new research and concepts in practical ways. Much of the published research available in journals in print and online has been through the ‘peer review’ process, where a draft of the manuscript proceeds through several rounds of reviews by at least one, usually anonymous, academic research peer specializing in that particular research field <span class="citation" data-cites="9GeeRsSN">[<a href="#ref-9GeeRsSN" role="doc-biblioref">14</a>]</span>. In theory, this ensures that all peer-reviewed published research is of a high quality, has been thoroughly proof-checked and is being seen by the most appropriate and wide reaching audience possible. Nevertheless, the process of publishing and reviewing scientific research is often daunting to students and early career researchers, and there is relatively little cohesive advice or guidelines about the process, often leaving such individuals to ‘figure it out for themselves’. In this section, we aim to provide a short overview of the process, both for authors and peer-reviewers, to eliminate some of the confusion and mystery about the publication of academic research.</p>
<h3 id="determining-coauthorship">Determining coauthorship</h3>
<p>One key thing to decide before beginning the manuscript submission process is who should be listed as an author of the paper <span class="citation" data-cites="1Dmi4hTux 1AiJvcSSw">[<a href="#ref-1Dmi4hTux" role="doc-biblioref">15</a>,<a href="#ref-1AiJvcSSw" role="doc-biblioref">16</a>]</span>. Scientific work is rarely conducted alone, and successful, meaningful ecological research often involves large collaborative efforts between teams of researchers, conservationists, land-owners, companies, technicians and students. Thus, determining who qualifies to be listed as a co-author is not always straight-forward. Generally speaking, an un-written rule is that an individual should be listed as a co-author if they contribute to at least two crucial components of the work, which may include but are not limited to: initial study design and conceptualization, obtaining funding, project management, field or laboratory work, data collating and processing, statistical analysis and manuscript drafting or proof-reading (Rahman et al., 2017). More often now journals require an ‘Author Contribution’ statement to be added to the manuscript or the online submission form, indicating the contribution of each author to the key tasks involved in compiling the manuscript (Rennie et al., 2000) (see Box 1 for an example author contribution statement).</p>
<p>Another potentially controversial and confusing decision when submitting multi-authored manuscripts is in what order should the authors appear in the author list (Tscharntke et al., 2007). The first author position, receiving the most credit for the work, is generally the individual who has played the most crucial role in the work (e.g. the project lead or primary writer), with all subsequent authors listed either in order of decreasing contribution, alphabetically or in order of seniority. In some scientific fields or for particular journals, the last author position is seen as prestigiously equal to first author position and is reserved for the team leader e.g. the project manager or research principal investigator. It is important to check if the chosen journal has specific instructions about authorship contributions and order to ensure that each individual receives the appropriate credit for their work on that research project.</p>
<center>
<figure>
<img src="https://user-images.githubusercontent.com/69354870/199533381-7fdf36f0-db36-443d-b60d-753b0a10e448.png" alt="Box1" />
<figcaption aria-hidden="true">Box1</figcaption>
</figure>
</center>
<p>The final decision to make is who acts as the corresponding author (e.g. the individual who submits the final manuscript draft and is responsible for handling all correspondence with the journal and editors). Usually this is the first (or last if appropriate) author of the paper, and is the person responsible for handling all queries, revisions and approving article proofs from the journal. The corresponding author is also committed to declaring on behalf of all co-authors that there are no conflict of interests or prior publications of the work, and that all work is original and was conducted ethically (Shewan and Coats, 2010).</p>
<h3 id="choosing-the-best-publication-outlet">Choosing the best publication outlet</h3>
<p>Once a draft of the manuscript has been completed and authorship has been decided, there is relatively little formal guidance other than that provided by colleagues as to the most appropriate avenue and format for publication. Modern publication today occurs primarily online, with the internet facilitating easy and quick access to thousands of publications and journals worldwide. However, each journal will differ in its impact, scope, relevance, target audience, and overall publication process (Lewallen and Crane, 2010; ). In Figure X, we present a brief overview of the main steps to consider when choosing a journal, each of which can present a potential hurdle in the publication process. Following these steps will ensure minimal time is wasted for both authors and journal editors in the submission and initial review stages, and that the manuscript will be directed to the most appropriate journal with the best chance of making it to peer-review.</p>
<p>A logical approach to choosing the best publication avenue is to start by asking whether the scope of the journal (i.e. the topics and types of research published) encompass that of your work (Step 1 in Figure X). This includes determining whether the journal publishes material targeted at specific areas of interest or publishes more broadly; some journals (e.g. Global Ecology and Biogeography, Journal of Applied Ecology or PLOS One) publish on a broad range of ecological, evolutionary, environmental, or biogeographical topics, but others target specific taxa (e.g. Fungal Diversity, Mammalian Biology etc.), methods (e.g. Environmental DNA or Remote Sensing in Ecology and Conservation), concepts (Global Food Security, Nature Sustainability) or geographical areas (e.g. European Journal of Environmental Sciences, Journal of Asia-Pacific Biodiversity). Reading the aims, scope or overview of the journal, as well as checking out their recently published papers, will provide an indication of whether your manuscript is likely to be a fit for the journal, and just as importantly, whether the journals’ target audience is appropriate for your research.</p>
<p>If your work fits within the scope of the journal, the second step might be to see if your type of manuscript matches those that they publish (Step 2 in Figure X). Before beginning any draft manuscript it is important to decide, in collaboration with the co-authors, what type of research you are aiming to present and publish, whether that is an original research article (most common), a review or synthesis paper, technical note, letter, response to another article or perhaps even a gallery of photos or uploads of raw data. Not all types of submission will be accepted by every journal. In addition, formatting guidelines for each journal (e.g. length of manuscript, headings, citation style etc.) often differ between different types of research, so making sure your work is correctly formatted per the manuscript type will be an important step in getting your manuscript sent out quickly for peer-review.</p>
<p>Additionally, not all journals have equal reach in terms of the number of readers, article quality or reach, and several metrics exist for authors to compare journals. Although controversial and often confusing, the main metric widely used is the Impact Factor (IF), calculated as the average number of times articles in a journal from the past two years have been cited in relation to the total number of articles published over the same two year period (Garfield, 1999). Although only useful when comparing journals, the IF provides an idea of the quality and reach of the journal, and provides an indication how likely your work will be accepted by the journal. Other metrics include the SCImago Journal Rank (SJR), taking into account the ‘value’ of a citation and journal prestige, and the Source Normalized Impact per Paper (SNIP), which measures citation impact within a subject field (cite; cite).</p>
<p>A final, yet by no means insignificant factor, are the fees, usually termed the Article Processing Charges (APCs), charged to the authors by the journal to publish your work (Figure X). Journal revenues generally comes from two sources: 1) publishing charges if the article is Open Access (OA) (available for anyone to read) or 2) journal subscription fees from individuals, libraries or organizations. If the latter, usually your work is only available to those with subscription access, potentially limiting your works’ reach and accessibility. However, OA publishing fees can be substantial, so being aware early on of the sources of funding for publication is important. Some large organizations e.g. universities might have waiver agreements with the larger OA publishers like Wiley, and additional financial support might be offered for authors from developing countries or other organizations, but often funding for publication will be the responsibility of the researcher, principal investigator or research team. Make sure to clarify funding sources for publication with your funder, supervisor or department early in the research process and always check out the journal guidelines for information on fees and waivers before starting a submission.</p>
<h3 id="who-should-review-your-manuscript">Who should review your manuscript?</h3>
<p>Obtaining feedback from colleagues and peers has been shown to be extremely valuable for producing cohesive, well-written and clear pieces of scientific writing (Paulus, 1999; Nicol et al., 2014). Right from the start of the initial design phase of a field study, experiment or piece of written work, asking for and implementing constructive criticism and recommendations at all stages in the process will massively improve the chance that your research will be successful, and will thus make a valuable contribution to the field when it comes to writing up the final manuscript for publication. Before the final manuscript submission, consider asking your supervisor, senior colleagues, peers or other collaborators not directly involved in the work, for their critical opinions throughout the whole of the research and writing process. This should help to identify and smooth out potential criticisms from the formal peer-review process, and will allow larger problems to be identified that might result in an outright manuscript rejection.</p>
<p>During the actual submission process however, a journal will sometimes request or even mandate authors to suggest the names of potential peer-reviewers (Moore et al., 2011; Liang, 2018). This saves time for the editors in finding potential reviewers specialized in that field, and is therefore also beneficial for the authors in receiving a timely response to their submission. If requested to do so, have a careful think about who might be suitable to review your work; these selected individuals should have no conflict of interest with your research, and should not have been involved with any stages of work, reviewing or writing the manuscript. Sensible suggestions most likely to be accepted by an editor will most likely be individuals with a strong publication record relevant to the manuscript e.g. someone you have cited frequently in your work, or a known expert in that field. Avoid if possible individuals from the same institution or geographic area, or very highly renowned individuals who may not be able to spare time for peer-review (Eyers, 2021). Less frequently, there may also be an option to suggest peer-reviewers to avoid: these might be individuals with conflicts of interest in your work, or you might want to avoid them if they are known to be a very harsh, critical reviewer or for some other personal reason. Similar rules apply if the journal asks you to list preferred or non-preferred editors: make sure to take the time to research the individuals and their areas of expertise to make sure that your manuscript lands in the hands of the person most likely to facilitate its procession through the peer-review process.</p>
<center>
<figure>
<img src="https://user-images.githubusercontent.com/69354870/199533472-b46b60c7-7da1-409a-8c58-6a329313560a.png" alt="Figure X. General outline of the four main steps to consider when choosing the most appropriate outlet for the publication of your research." />
<figcaption aria-hidden="true">Figure X. General outline of the four main steps to consider when choosing the most appropriate outlet for the publication of your research.</figcaption>
</figure>
</center>
<h3 id="how-to-constructively-provide-feedback-as-a-reviewer">How to constructively provide feedback as a reviewer?</h3>
<p>Getting involved in the peer-review process is a crucial, rewarding component to being a successful academic or researcher. Becoming a peer-reviewer for a journal offers opportunities to be exposed to the most recent scientific developments in your field, as well as the chance to develop critical thinking skills and the ability to rapidly read and understand vast quantities of scientific literature. At the same time, journals rely heavily on having a significant bank of reviewers to help them examine large numbers of submissions to decide which are most noteworthy and suitable for publication (Lubek, 2018). In general, the expectations of a reviewer are to provide robust, in-depth feedback in a timely fashion on research within their field of expertise, in which they might reasonably consider themselves an expert or specialist. A reviewer should act professionally at all times by declaring any conflicts of interests, maintaining good communication with the editorial team and making sure all review material is kept confidential (Carter, 2008; Ramsden et al., 2014).</p>
<p>Anyone can become a peer-reviewer, with the chances of being invited to review by the journal increasing generally with the number of publications and citations to your name (e.g. your academic experience and knowledge within that particular field), and the reach of your online academic research profile. However, even those with no publication experience can become reviewers, perhaps by asking for recommendation from a colleague, directly approaching the journal or editors, or volunteering to conduct peer-reviews for a senior researcher. It is now also possible to get recognition and verification for peer-review through online sites and services such as the Web of Science Reviewer Recognition Service (previously Publons), Elsevier’s Reviewer Hub or ORCID, and many subscription journals offer free temporary subscriptions to that journal as a reward. Building up your online peer-review history may then facilitate further invitations from other journals.</p>
<p>Once an invite is issued by the journal editor and accepted, the reviewer is able to see the full material and begin their review. Starting a review is often overwhelming, especially for first time reviewers, but there are many resources to help with this, including usually quite detailed advice from the journal online aimed at reviewers. Sometimes the journal also provides a more structured review template, consisting of a series of short question pertaining to the material to help direct the review. However, as this is not always available, we have compiled a brief step-by-step guide to carrying out a systematic review, based on online advice from three of the major academic publishers; Wiley, PLOS and ELSEVIER (Figure X2). We acknowledge that there are many different ways to carry out a good review, but we hope this schematic will provide a useful starting point for first time reviewers to help them develop their own review styles.</p>
<center>
<figure>
<img src="https://user-images.githubusercontent.com/69354870/199533596-ec5e4666-3390-417b-815e-e87a279c5712.png" alt="Figure X2. A brief step-by-step example guide to carrying out a systematic unstructured peer review. Compiled from online informational sources by major publishers Wiley, PLOS and Elsevier." />
<figcaption aria-hidden="true">Figure X2. A brief step-by-step example guide to carrying out a systematic unstructured peer review. Compiled from online informational sources by major publishers Wiley, PLOS and Elsevier.</figcaption>
</figure>
</center>
<h3 id="responding-to-reviews">Responding to reviews</h3>
<p>Once a manuscript is submitted and has undergone an initial proofing process, the assigned journal editor will then look for suitable peer-reviewers to send the manuscript to for review. A manuscript will generally be seen by 1-3 peer-reviewers who will feedback their comments and recommendations (accept, accept with revision or reject) to the editor. The editor will then send their own comments and recommendations along with those of the reviewers back to the authors letting them know the outcome and next steps. It is rare for a paper to be accepted outright, with the most likely positive outcome either a major or minor revision depending on the number and severity of problems with the work. If this is the case, the authors then have the opportunity to respond in writing and to revise their work to be reconsidered for publication.</p>
<p>A good response should generally include: 1) acknowledgement of the time and effort spent on the reviews by the editor(s) and reviewer(s), 2) specific, detailed point-by-point responses to each individual comment, and 3) clear indications of all the changes, additions and deletions made to the manuscript and supplementary materials, including the use of specific page and line number references and tracked changes in documents where possible. If a specific template is not provided by the journal, a good format for the response is a letter addressed to the assigned editor if known, or editor-in-chief of the journal as outlined in Box 2. The letter includes key identifying information for the manuscript and begins by thanking the editor and reviewers for the review before leading into addressing all general and specific responses one by one.</p>
<p>When responding to each comment remember that an editor’s time is valuable and limited (most journal editors are part-time volunteers), so use language that is precise, concise and does not repeat information already in, or being added to the manuscript. To give an example, if a reviewer suggests expanding the literature review in a particular area, a good response might be: “We have added additional citations and explanation of each study at Line 239-250”. A bad response might thus be: “We did more research into this area and found this paper by AB that states this…, and we also have added this paper by CD et al. that says this…, and this one by EF supporting our argument here where we state… etc.”. All that is required is a clear response acknowledging the comment and showing if, how and where you have made changes.</p>
<p>One important final thing to mention is that authors actually have no obligation to carry out all or even any of the corrections if they fundamentally disagree with the comments, they are outside the scope of the work or the work has been misinterpreted or misunderstood. If so, it is perfectly acceptable to write a response clarifying why you disagree or do not wish to make the suggested change. Many students and early-career researchers often feel underqualified to disagree with a reviewer, but remember that you conducted the work and should have a good understanding of the feasibility and robustness of your methodology, and will likely have considered the limitations and scope of your study for much longer than the reviewer. If writing in disagreement, just make sure to always acknowledge the point of view of the reviewer, and always use very clear examples and cite supporting literature or other studies to back up your point of view. As with writing a peer-review (see Figure X2), always using courteous and professional language, and try to avoid sounding passive-aggressive or even aggressive, even if the reviewer is unnecessarily critical or harsh.</p>
<center>
<figure>
<img src="https://user-images.githubusercontent.com/69354870/199533650-ddecdd2d-aaf2-47bc-b5bd-eafbd4d42c7a.png" alt="Box2" />
<figcaption aria-hidden="true">Box2</figcaption>
</figure>
</center>
<h3 id="marketing-your-manuscript-after-acceptance">Marketing your manuscript after acceptance</h3>
<p>The peer-review process can be lengthy depending on the number of reviewers, rounds of revision, and journal popularity, and it is not unusual for several months to even a year to pass between first submission and final acceptance and publication. However, most journals have online tracking systems to keep you up to date with the status of your manuscript (e.g. under review, pending decision, accepted etc.), and in the meantime it is important to think about how you wish to promote your work once it is published. As mentioned previously, it is not just the scientific community who will be interested in your work, and your findings may have broad ecological or conservation applications. Knowing how and where to communicate your research to the broader community will likely involve the use of other platforms including online social media, research collation sites and more mainstream media outlets.</p>
<p>Having an established, professional social media presence on platforms such as Twitter, LinkedIn, Facebook and Instagram to name but a few, facilitates immediate, free marketing of your work, and provides easy links and ways for others to share things they find interesting. Consider how to effectively use relevant images, videos, short captions, hash-tags and handles of your co-authors to spread the broad message of your research to your target audience. Other more research specific online platforms such as Research Gate and ORCID allow for you to collate and update your publication and peer-review record, and provides ways to follow and connect with other fellow researchers in your field. You may also wish to collaborate with the advertising or marketing department within your company or academic institution to create a more general press release to traditional media outlets (radio, TV, magazines, newspapers etc.); this type of promotion can dramatically increase the reach of your work and lead to many more opportunities to disseminate your work to those that will be best to practically implement it. Planning ahead and preparing the appropriate materials to assist with your research promotion alongside your co-authors will be key to successfully making the most of these strategies. Remember that a lot of work (often years!) will have gone into designing, conducting and writing up this research, so it is important that it, and all the authors and other collaborators, get the recognition they deserve!</p>
<h3 id="references-1">References</h3>
<ul>
<li>Allen, L., Scott, J., Brand, A., Hlava, M., Altman, M., 2014. Publishing: Credit where credit is due. Nature 508, 312–313.</li>
<li>Bornmann, L., 2011. Scientific peer review. Annu. Rev. Inf. Sci. Technol. 45, 197–245.</li>
<li>Carter, V.K., 2008. Five Steps to Becoming a Better Peer Reviewer. Coll. Teach. 56, 85–88.</li>
<li>Eyers, C.E., 2021. “What’s in a Name?”: Considerations for Identifying Preferred Reviewers. J. Proteome Res. 20, 4625–4626.</li>
<li>Garfield, E., 1999. Journal impact factor: a brief review. CMAJ 161, 979–980.</li>
<li>Lewallen, L.P., Crane, P.B., 2010. Choosing a Publication Venue. J. Prof. Nurs. 26, 250–254.</li>
<li>Liang, Y., 2018. Should authors suggest reviewers? A comparative study of the performance of author-suggested and editor-selected reviewers at a biological journal. Learn. Publ. 31, 216–221.</li>
<li>Lubek, J., 2018. The important role of the peer reviewer. Oral Surg. Oral Med. Oral Pathol. Oral Radiol. 125, 391–392.</li>
<li>Moore, J.L., Neilson, E.G., Siegel, V., Nephrology, on behalf of the A.E. at the J. of the A.S. of, 2011. Effect of Recommendations from Reviewers Suggested or Excluded by Authors. J. Am. Soc. Nephrol. 22, 1598–1602.</li>
<li>Nicol, D., Thomson, A., Breslin, C., 2014. Rethinking feedback practices in higher education: a peer review perspective. Assess. Eval. High. Educ. 39, 102–122.</li>
<li>Paulus, T.M., 1999. The effect of peer and teacher feedback on student writing. J. Second Lang. Writ. 8, 265–289.</li>
<li>Rahman, M.T., Regenstein, J.M., Kassim, N.L.A., Haque, N., 2017. The need to quantify authors’ relative intellectual contributions in a multi-author paper. J. Informetr. 11, 275–281.</li>
<li>Ramsden, V.R., Pimlott, N., Woollard, R., Kvern, B., Handford, C., Dunikowski, L., Gagnon, S., Hoffart, K., Ladouceur, R., Lambert, Y., 2014. Becoming a peer reviewer: Engaging in sharing and gaining knowledge. Can. Fam. Physician 60, 1158–1160.</li>
<li>Rennie, D., Flanagin, A., Yank, V., 2000. The Contributions of Authors. JAMA 284, 89–91.</li>
<li>Shewan, L.G., Coats, A.J.S., 2010. Ethics in the authorship and publishing of scientific articles. Int. J. Cardiol. 144, 1–2.</li>
<li>Tscharntke, T., Hochberg, M.E., Rand, T.A., Resh, V.H., Krauss, J., 2007. Author Sequence and Credit for Contributions in Multiauthored Publications. PLOS Biol. 5, e18.</li>
</ul>
<h3 id="other-useful-links">Other Useful Links</h3>
<ul>
<li>https://authorservices.wiley.com/Reviewers/journal-reviewers/how-to-perform-a-peer-review/step-by-step-guide-to-reviewing-a-manuscript.html</li>
<li>https://plos.org/resource/how-to-write-a-peer-review/</li>
<li>https://www.elsevier.com/reviewers/how-to-review</li>
</ul>
<h2 class="page_break_before" id="figures">Figures</h2>
<p>![<img src="images/OpenWorkflows_Chart.png" id="fig:openworkflow-diagram" alt="Schematic overview of the components of open and accessible workflows for wildlife research and decision-making." /></p>
<!--Reference this figure in text like this (Figure @fig:openworkflow-diagram).-->
<h2 id="tables">Tables</h2>
<h2 class="page_break_before" id="references-2">References</h2>
<!-- Explicitly insert bibliography here -->
<div id="refs" class="references csl-bib-body" role="doc-bibliography">
<div id="ref-vbxx8YaD" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">1. </div><div class="csl-right-inline"><strong>Toward rigorous use of expert knowledge in ecological research</strong> <div class="csl-block">M Drescher, AH Perera, CJ Johnson, LJ Buse, CA Drew, MA Burgman</div> <em>Ecosphere</em> (2013-07) <a href="https://doi.org/gg5nfq">https://doi.org/gg5nfq</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1890/es12-00415.1">10.1890/es12-00415.1</a></div></div>
</div>
<div id="ref-1GZya86fQ" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">2. </div><div class="csl-right-inline"><strong>What do we know about public attitudes towards experts? Reviewing survey data in the United Kingdom and European Union</strong> <div class="csl-block">Katharine Dommett, Warren Pearce</div> <em>Public Understanding of Science</em> (2019-05-31) <a href="https://doi.org/gsdnr2">https://doi.org/gsdnr2</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1177/0963662519852038">10.1177/0963662519852038</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/31151371">31151371</a></div></div>
</div>
<div id="ref-juUi38G9" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">3. </div><div class="csl-right-inline"><strong>Ecology for transformation</strong> <div class="csl-block">Stephen R Carpenter, Carl Folke</div> <em>Trends in Ecology &amp;amp; Evolution</em> (2006-06) <a href="https://doi.org/ds4j4v">https://doi.org/ds4j4v</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1016/j.tree.2006.02.007">10.1016/j.tree.2006.02.007</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/16769430">16769430</a></div></div>
</div>
<div id="ref-1HhzKAC1K" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">4. </div><div class="csl-right-inline"><strong>Improving the culture of interdisciplinary collaboration in ecology by expanding measures of success</strong> <div class="csl-block">Simon J Goring, Kathleen C Weathers, Walter K Dodds, Patricia A Soranno, Lynn C Sweet, Kendra S Cheruvelil, John S Kominoski, Janine Rüegg, Alexandra M Thorn, Ryan M Utz</div> <em>Frontiers in Ecology and the Environment</em> (2014-02) <a href="https://doi.org/rb2">https://doi.org/rb2</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1890/120370">10.1890/120370</a></div></div>
</div>
<div id="ref-OdGVIIR2" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">5. </div><div class="csl-right-inline"><strong>Creating and maintaining high‐performing collaborative research teams: the importance of diversity and interpersonal skills</strong> <div class="csl-block">Kendra S Cheruvelil, Patricia A Soranno, Kathleen C Weathers, Paul C Hanson, Simon J Goring, Christopher T Filstrup, Emily K Read</div> <em>Frontiers in Ecology and the Environment</em> (2014-02) <a href="https://doi.org/f52rtd">https://doi.org/f52rtd</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1890/130001">10.1890/130001</a></div></div>
</div>
<div id="ref-BQhgQLjo" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">6. </div><div class="csl-right-inline"><strong>Ecological research and environmental management: We need different interfaces based on different knowledge types</strong> <div class="csl-block">Frédéric Gosselin, Thomas Cordonnier, Isabelle Bilger, Marielle Jappiot, Christophe Chauvin, Marion Gosselin</div> <em>Journal of Environmental Management</em> (2018-07) <a href="https://doi.org/gdp8w8">https://doi.org/gdp8w8</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1016/j.jenvman.2018.04.025">10.1016/j.jenvman.2018.04.025</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/29704834">29704834</a></div></div>
</div>
<div id="ref-HrWLvwSo" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">7. </div><div class="csl-right-inline"><strong>Ecologist engagement in translational science is imperative for building resilience to global change threats</strong> <div class="csl-block">Kennedy Rubert-Nason, AM Aramati Casper, Matt Jurjonas, Caitlin Mandeville, Rebecca Potter, Kirsten Schwarz</div> <em>Rethinking Ecology</em> (2021-04-23) <a href="https://doi.org/gsdnr3">https://doi.org/gsdnr3</a> <div class="csl-block">DOI: <a href="https://doi.org/10.3897/rethinkingecology.6.64103">10.3897/rethinkingecology.6.64103</a></div></div>
</div>
<div id="ref-bF8TZ7uw" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">8. </div><div class="csl-right-inline"><strong>The history of public participation in ecological research</strong> <div class="csl-block">Abraham Miller-Rushing, Richard Primack, Rick Bonney</div> <em>Frontiers in Ecology and the Environment</em> (2012-08) <a href="https://doi.org/gf5tv7">https://doi.org/gf5tv7</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1890/110278">10.1890/110278</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/37013221">37013221</a></div></div>
</div>
<div id="ref-PI3uanHz" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">9. </div><div class="csl-right-inline"><strong>Citizen science can improve conservation science, natural resource management, and environmental protection</strong> <div class="csl-block">Duncan C McKinley, Abe J Miller-Rushing, Heidi L Ballard, Rick Bonney, Hutch Brown, Susan C Cook-Patton, Daniel M Evans, Rebecca A French, Julia K Parrish, Tina B Phillips, … Michael A Soukup</div> <em>Biological Conservation</em> (2017-04) <a href="https://doi.org/f96qkt">https://doi.org/f96qkt</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1016/j.biocon.2016.05.015">10.1016/j.biocon.2016.05.015</a></div></div>
</div>
<div id="ref-17087XSCD" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">10. </div><div class="csl-right-inline"><strong>The Importance of Replication in Wildlife Research</strong> <div class="csl-block">Douglas H Johnson</div> <em>The Journal of Wildlife Management</em> (2002-10) <a href="https://doi.org/bv6mx5">https://doi.org/bv6mx5</a> <div class="csl-block">DOI: <a href="https://doi.org/10.2307/3802926">10.2307/3802926</a></div></div>
</div>
<div id="ref-166u7Adqx" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">11. </div><div class="csl-right-inline"><strong>Pseudoreplication and the Design of Ecological Field Experiments</strong> <div class="csl-block">Stuart H Hurlbert</div> <em>Ecological Monographs</em> (1984-06) <a href="https://doi.org/c37ccj">https://doi.org/c37ccj</a> <div class="csl-block">DOI: <a href="https://doi.org/10.2307/1942661">10.2307/1942661</a></div></div>
</div>
<div id="ref-16oQ8DJLK" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">12. </div><div class="csl-right-inline"><strong>Pseudoreplication Revisited</strong> <div class="csl-block">Robert A Heffner, Mark J Butler, Colleen Keelan Reilly</div> <em>Ecology</em> (1996-12) <a href="https://doi.org/cz2tzk">https://doi.org/cz2tzk</a> <div class="csl-block">DOI: <a href="https://doi.org/10.2307/2265754">10.2307/2265754</a></div></div>
</div>
<div id="ref-1ElVE5fpk" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">13. </div><div class="csl-right-inline"><strong>TEMPORAL AND SPATIAL VARIATION IN ENVIRONMENTAL IMPACT ASSESSMENT</strong> <div class="csl-block">Allan Stewart-Oaten, James R Bence</div> <em>Ecological Monographs</em> (2001-05) <a href="https://doi.org/dmx7gm">https://doi.org/dmx7gm</a> <div class="csl-block">DOI: <a href="https://doi.org/10.2307/2657220">10.2307/2657220</a></div></div>
</div>
<div id="ref-9GeeRsSN" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">14. </div><div class="csl-right-inline"><strong>Scientific peer review</strong> <div class="csl-block">Lutz Bornmann</div> <em>Annual Review of Information Science and Technology</em> (2011) <a href="https://doi.org/ggqp56">https://doi.org/ggqp56</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1002/aris.2011.1440450112">10.1002/aris.2011.1440450112</a></div></div>
</div>
<div id="ref-1Dmi4hTux" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">15. </div><div class="csl-right-inline"><strong>Publishing: Credit where credit is due</strong> <div class="csl-block">Liz Allen, Jo Scott, Amy Brand, Marjorie Hlava, Micah Altman</div> <em>Nature</em> (2014-04-16) <a href="https://doi.org/sdf">https://doi.org/sdf</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1038/508312a">10.1038/508312a</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/24745070">24745070</a></div></div>
</div>
<div id="ref-1AiJvcSSw" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">16. </div><div class="csl-right-inline"><strong>The need to quantify authors’ relative intellectual contributions in a multi-author paper</strong> <div class="csl-block">Mohammad Tariqur Rahman, Joe Mac Regenstein, Noor Lide Abu Kassim, Nazmul Haque</div> <em>Journal of Informetrics</em> (2017-02) <a href="https://doi.org/f9twhv">https://doi.org/f9twhv</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1016/j.joi.2017.01.002">10.1016/j.joi.2017.01.002</a></div></div>
</div>
</div>
<!-- default theme -->

<style>
  /* import google fonts */
  @import url("https://fonts.googleapis.com/css?family=Open+Sans:400,600,700");
  @import url("https://fonts.googleapis.com/css?family=Source+Code+Pro");

  /* -------------------------------------------------- */
  /* global */
  /* -------------------------------------------------- */

  /* all elements */
  * {
    /* force sans-serif font unless specified otherwise */
    font-family: "Open Sans", "Helvetica", sans-serif;

    /* prevent text inflation on some mobile browsers */
    -webkit-text-size-adjust: none !important;
    -moz-text-size-adjust: none !important;
    -o-text-size-adjust: none !important;
    text-size-adjust: none !important;
  }

  @media only screen {
    /* "page" element */
    body {
      position: relative;
      box-sizing: border-box;
      font-size: 12pt;
      line-height: 1.5;
      max-width: 8.5in;
      margin: 20px auto;
      padding: 40px;
      border-radius: 5px;
      border: solid 1px #bdbdbd;
      box-shadow: 0 0 20px rgba(0, 0, 0, 0.05);
      background: #ffffff;
    }
  }

  /* when on screen < 8.5in wide */
  @media only screen and (max-width: 8.5in) {
    /* "page" element */
    body {
      padding: 20px;
      margin: 0;
      border-radius: 0;
      border: none;
      box-shadow: 0 0 20px rgba(0, 0, 0, 0.05) inset;
      background: none;
    }
  }

  /* -------------------------------------------------- */
  /* headings */
  /* -------------------------------------------------- */

  /* all headings */
  h1,
  h2,
  h3,
  h4,
  h5,
  h6 {
    margin: 20px 0;
    padding: 0;
    font-weight: bold;
  }

  /* biggest heading */
  h1 {
    margin: 40px 0;
    text-align: center;
  }

  /* second biggest heading */
  h2 {
    margin-top: 30px;
    padding-bottom: 5px;
    border-bottom: solid 1px #bdbdbd;
  }

  /* heading font sizes */
  h1 {
    font-size: 2em;
  }
  h2 {
    font-size: 1.5em;
  }
  h3 {
    font-size: 1.35em;
  }
  h4 {
    font-size: 1.25em;
  }
  h5 {
    font-size: 1.15em;
  }
  h6 {
    font-size: 1em;
  }

  /* -------------------------------------------------- */
  /* manuscript header */
  /* -------------------------------------------------- */

  /* manuscript title */
  header > h1 {
    margin: 0;
  }

  /* manuscript title caption text (ie "automatically generated on") */
  header + p {
    text-align: center;
    margin-top: 10px;
  }

  /* -------------------------------------------------- */
  /* text elements */
  /* -------------------------------------------------- */

  /* links */
  a {
    color: #2196f3;
    overflow-wrap: break-word;
  }

  /* superscripts and subscripts */
  sub,
  sup {
    /* prevent from affecting line height */
    line-height: 0;
  }

  /* unordered and ordered lists*/
  ul,
  ol {
    padding-left: 20px;
  }

  /* class for styling text semibold */
  .semibold {
    font-weight: 600;
  }

  /* class for styling elements horizontally left aligned */
  .left {
    display: block;
    text-align: left;
    margin-left: auto;
    margin-right: 0;
    justify-content: left;
  }

  /* class for styling elements horizontally centered */
  .center {
    display: block;
    text-align: center;
    margin-left: auto;
    margin-right: auto;
    justify-content: center;
  }

  /* class for styling elements horizontally right aligned */
  .right {
    display: block;
    text-align: right;
    margin-left: 0;
    margin-right: auto;
    justify-content: right;
  }

  /* -------------------------------------------------- */
  /* section elements */
  /* -------------------------------------------------- */

  /* horizontal divider line */
  hr {
    border: none;
    height: 1px;
    background: #bdbdbd;
  }

  /* paragraphs, horizontal dividers, figures, tables, code */
  p,
  hr,
  figure,
  table,
  pre {
    /* treat all as "paragraphs", with consistent vertical margins */
    margin-top: 20px;
    margin-bottom: 20px;
  }

  /* -------------------------------------------------- */
  /* figures */
  /* -------------------------------------------------- */

  /* figure */
  figure {
    max-width: 100%;
    margin-left: auto;
    margin-right: auto;
  }

  /* figure caption */
  figcaption {
    padding: 0;
    padding-top: 10px;
  }

  /* figure image element */
  figure > img,
  figure > svg {
    max-width: 100%;
    display: block;
    margin-left: auto;
    margin-right: auto;
  }

  /* figure auto-number */
  img + figcaption > span:first-of-type,
  svg + figcaption > span:first-of-type {
    font-weight: bold;
    margin-right: 5px;
  }

  /* -------------------------------------------------- */
  /* tables */
  /* -------------------------------------------------- */

  /* table */
  table {
    border-collapse: collapse;
    border-spacing: 0;
    width: 100%;
    margin-left: auto;
    margin-right: auto;
  }

  /* table cells */
  th,
  td {
    border: solid 1px #bdbdbd;
    padding: 10px;
    /* squash table if too wide for page by forcing line breaks */
    overflow-wrap: break-word;
    word-break: break-word;
  }

  /* header row and even rows */
  th,
  tr:nth-child(2n) {
    background-color: #fafafa;
  }

  /* odd rows */
  tr:nth-child(2n + 1) {
    background-color: #ffffff;
  }

  /* table caption */
  caption {
    text-align: left;
    padding: 0;
    padding-bottom: 10px;
  }

  /* table auto-number */
  table > caption > span:first-of-type {
    font-weight: bold;
    margin-right: 5px;
  }

  /* -------------------------------------------------- */
  /* code */
  /* -------------------------------------------------- */

  /* multi-line code block */
  pre {
    padding: 10px;
    background-color: #eeeeee;
    color: #000000;
    border-radius: 5px;
    break-inside: avoid;
    text-align: left;
  }

  /* inline code, ie code within normal text */
  :not(pre) > code {
    padding: 0 4px;
    background-color: #eeeeee;
    color: #000000;
    border-radius: 5px;
  }

  /* code text */
  /* apply all children, to reach syntax highlighting sub-elements */
  code,
  code * {
    /* force monospace font */
    font-family: "Source Code Pro", "Courier New", monospace;
  }

  /* -------------------------------------------------- */
  /* quotes */
  /* -------------------------------------------------- */

  /* quoted text */
  blockquote {
    margin: 0;
    padding: 0;
    border-left: 4px solid #bdbdbd;
    padding-left: 16px;
    break-inside: avoid;
  }

  /* -------------------------------------------------- */
  /* banners */
  /* -------------------------------------------------- */

  /* info banners */
  .banner {
    box-sizing: border-box;
    display: block;
    position: relative;
    width: 100%;
    margin-top: 20px;
    margin-bottom: 20px;
    padding: 20px;
    text-align: center;
  }

  /* paragraph in banner */
  .banner > p {
    margin: 0;
  }

  /* -------------------------------------------------- */
  /* highlight colors */
  /* -------------------------------------------------- */

  .white {
    background: #ffffff;
  }
  .lightgrey {
    background: #eeeeee;
  }
  .grey {
    background: #757575;
  }
  .darkgrey {
    background: #424242;
  }
  .black {
    background: #000000;
  }
  .lightred {
    background: #ffcdd2;
  }
  .lightyellow {
    background: #ffecb3;
  }
  .lightgreen {
    background: #dcedc8;
  }
  .lightblue {
    background: #e3f2fd;
  }
  .lightpurple {
    background: #f3e5f5;
  }
  .red {
    background: #f44336;
  }
  .orange {
    background: #ff9800;
  }
  .yellow {
    background: #ffeb3b;
  }
  .green {
    background: #4caf50;
  }
  .blue {
    background: #2196f3;
  }
  .purple {
    background: #9c27b0;
  }
  .white,
  .lightgrey,
  .lightred,
  .lightyellow,
  .lightgreen,
  .lightblue,
  .lightpurple,
  .orange,
  .yellow,
  .white a,
  .lightgrey a,
  .lightred a,
  .lightyellow a,
  .lightgreen a,
  .lightblue a,
  .lightpurple a,
  .orange a,
  .yellow a {
    color: #000000;
  }
  .grey,
  .darkgrey,
  .black,
  .red,
  .green,
  .blue,
  .purple,
  .grey a,
  .darkgrey a,
  .black a,
  .red a,
  .green a,
  .blue a,
  .purple a {
    color: #ffffff;
  }

  /* -------------------------------------------------- */
  /* buttons */
  /* -------------------------------------------------- */

  /* class for styling links like buttons */
  .button {
    display: inline-flex;
    justify-content: center;
    align-items: center;
    margin: 5px;
    padding: 10px 20px;
    font-size: 0.75em;
    font-weight: 600;
    text-transform: uppercase;
    text-decoration: none;
    letter-spacing: 1px;
    background: none;
    color: #2196f3;
    border: solid 1px #bdbdbd;
    border-radius: 5px;
  }

  /* buttons when hovered */
  .button:hover:not([disabled]),
  .icon_button:hover:not([disabled]) {
    cursor: pointer;
    background: #f5f5f5;
  }

  /* buttons when disabled */
  .button[disabled],
  .icon_button[disabled] {
    opacity: 0.35;
    pointer-events: none;
  }

  /* class for styling buttons containg only single icon */
  .icon_button {
    display: inline-flex;
    justify-content: center;
    align-items: center;
    text-decoration: none;
    margin: 0;
    padding: 0;
    background: none;
    border-radius: 5px;
    border: none;
    width: 20px;
    height: 20px;
    min-width: 20px;
    min-height: 20px;
  }

  /* icon button inner svg image */
  .icon_button > svg {
    height: 16px;
  }

  /* -------------------------------------------------- */
  /* icons */
  /* -------------------------------------------------- */

  /* class for styling icons inline with text */
  .inline_icon {
    height: 1em;
    position: relative;
    top: 0.125em;
  }

  /* -------------------------------------------------- */
  /* references */
  /* -------------------------------------------------- */

  .csl-entry {
    margin-top: 15px;
    margin-bottom: 15px;
  }

  /* -------------------------------------------------- */
  /* print control */
  /* -------------------------------------------------- */

  @media print {
    @page {
      /* suggested printing margin */
      margin: 0.5in;
    }

    /* document and "page" elements */
    html,
    body {
      margin: 0;
      padding: 0;
      width: 100%;
      height: 100%;
    }

    /* "page" element */
    body {
      font-size: 11pt !important;
      line-height: 1.35;
    }

    /* all headings */
    h1,
    h2,
    h3,
    h4,
    h5,
    h6 {
      margin: 15px 0;
    }

    /* figures and tables */
    figure,
    table {
      font-size: 0.85em;
    }

    /* table cells */
    th,
    td {
      padding: 5px;
    }

    /* shrink font awesome icons */
    i.fas,
    i.fab,
    i.far,
    i.fal {
      transform: scale(0.85);
    }

    /* decrease banner margins */
    .banner {
      margin-top: 15px;
      margin-bottom: 15px;
      padding: 15px;
    }

    /* class for centering an element vertically on its own page */
    .page_center {
      margin: auto;
      width: 100%;
      height: 100%;
      display: flex;
      align-items: center;
      vertical-align: middle;
      break-before: page;
      break-after: page;
    }

    /* always insert a page break before the element */
    .page_break_before {
      break-before: page;
    }

    /* always insert a page break after the element */
    .page_break_after {
      break-after: page;
    }

    /* avoid page break before the element */
    .page_break_before_avoid {
      break-before: avoid;
    }

    /* avoid page break after the element */
    .page_break_after_avoid {
      break-after: avoid;
    }

    /* avoid page break inside the element */
    .page_break_inside_avoid {
      break-inside: avoid;
    }
  }

  /* -------------------------------------------------- */
  /* override pandoc css quirks */
  /* -------------------------------------------------- */

  .sourceCode {
    /* prevent unsightly overflow in wide code blocks */
    overflow: auto !important;
  }

  div.sourceCode {
    /* prevent background fill on top-most code block  container */
    background: none !important;
  }

  .sourceCode * {
    /* force consistent line spacing */
    line-height: 1.5 !important;
  }

  div.sourceCode {
    /* style code block margins same as <pre> element */
    margin-top: 20px;
    margin-bottom: 20px;
  }

  /* -------------------------------------------------- */
  /* tablenos */
  /* -------------------------------------------------- */

  /* tablenos wrapper */
  .tablenos {
    width: 100%;
    margin: 20px 0;
  }

  .tablenos > table {
    /* move margins from table to table_wrapper to allow margin collapsing */
    margin: 0;
  }

  @media only screen {
    /* tablenos wrapper */
    .tablenos {
      /* show scrollbar on tables if necessary to prevent overflow */
      overflow-x: auto !important;
    }

    .tablenos th,
    .tablenos td {
      overflow-wrap: unset !important;
      word-break: unset !important;
    }

    /* table in wrapper */
    .tablenos table,
    .tablenos table * {
      /* don't break table words */
      overflow-wrap: normal !important;
    }
  }
</style>
<!-- 
    Plugin Core

    Functions needed for and shared across all first-party plugins.
-->

<script>
  // get element that is target of hash (from link element or url)
  function getHashTarget(link) {
    const hash = link ? link.hash : window.location.hash;
    const id = hash.slice(1);
    let target = document.querySelector(`[id="${id}"]`);
    if (!target) return;

    // if figure or table, modify target to get expected element
    if (id.indexOf("fig:") === 0) target = target.querySelector("figure");
    if (id.indexOf("tbl:") === 0) target = target.querySelector("table");

    return target;
  }

  // get position/dimensions of element or viewport
  function getRectInView(element) {
    let rect = {};
    rect.left = 0;
    rect.top = 0;
    rect.right = document.documentElement.clientWidth;
    rect.bottom = document.documentElement.clientHeight;
    let style = {};

    if (element instanceof HTMLElement) {
      rect = element.getBoundingClientRect();
      style = window.getComputedStyle(element);
    }

    const margin = {};
    margin.left = parseFloat(style.marginLeftWidth) || 0;
    margin.top = parseFloat(style.marginTopWidth) || 0;
    margin.right = parseFloat(style.marginRightWidth) || 0;
    margin.bottom = parseFloat(style.marginBottomWidth) || 0;

    const border = {};
    border.left = parseFloat(style.borderLeftWidth) || 0;
    border.top = parseFloat(style.borderTopWidth) || 0;
    border.right = parseFloat(style.borderRightWidth) || 0;
    border.bottom = parseFloat(style.borderBottomWidth) || 0;

    const newRect = {};
    newRect.left = rect.left + margin.left + border.left;
    newRect.top = rect.top + margin.top + border.top;
    newRect.right = rect.right + margin.right + border.right;
    newRect.bottom = rect.bottom + margin.bottom + border.bottom;
    newRect.width = newRect.right - newRect.left;
    newRect.height = newRect.bottom - newRect.top;

    return newRect;
  }

  // get position of element relative to page
  function getRectInPage(element) {
    const rect = getRectInView(element);
    const body = getRectInView(document.body);

    const newRect = {};
    newRect.left = rect.left - body.left;
    newRect.top = rect.top - body.top;
    newRect.right = rect.right - body.left;
    newRect.bottom = rect.bottom - body.top;
    newRect.width = rect.width;
    newRect.height = rect.height;

    return newRect;
  }

  // get closest element before specified element that matches query
  function firstBefore(element, query) {
    while (element && element !== document.body && !element.matches(query))
      element = element.previousElementSibling || element.parentNode;

    return element;
  }

  // check if element is part of collapsed heading
  function isCollapsed(element) {
    while (element && element !== document.body) {
      if (element.dataset.collapsed === "true") return true;
      element = element.parentNode;
    }
    return false;
  }

  // expand any collapsed parent containers of element if necessary
  function expandElement(element) {
    if (isCollapsed(element)) {
      // accordion plugin
      const heading = firstBefore(element, "h2");
      if (heading) heading.click();
      // details/summary HTML element
      const summary = firstBefore(element, "summary");
      if (summary) summary.click();
    }
  }

  // scroll to and focus element
  function goToElement(element, offset) {
    // expand accordion section if collapsed
    expandElement(element);
    const y =
      getRectInView(element).top -
      getRectInView(document.documentElement).top -
      (offset || 0);

    // trigger any function listening for "onscroll" event
    window.dispatchEvent(new Event("scroll"));
    window.scrollTo(0, y);
    document.activeElement.blur();
    element.focus();
  }

  // get list of elements after a start element up to element matching query
  function nextUntil(element, query, exclude) {
    const elements = [];
    while (((element = element.nextElementSibling), element)) {
      if (element.matches(query)) break;
      if (!element.matches(exclude)) elements.push(element);
    }
    return elements;
  }
</script>
<!--
  Accordion Plugin

  Allows sections of content under h2 headings to be collapsible.
-->

<script type="module">
  // whether to always start expanded ('false'), always start collapsed
  // ('true'), or start collapsed when screen small ('auto')
  const startCollapsed = "auto";

  // start script
  function start() {
    // run through each <h2> heading
    const headings = document.querySelectorAll("h2");
    for (const heading of headings) {
      addArrow(heading);

      // start expanded/collapsed based on option
      if (
        startCollapsed === "true" ||
        (startCollapsed === "auto" && isSmallScreen()) ||
        heading.dataset.collapsed === "true"
      )
        collapseHeading(heading);
      else expandElement(heading);
    }

    // attach hash change listener to window
    window.addEventListener("hashchange", onHashChange);
  }

  // when hash (eg manuscript.html#introduction) changes
  function onHashChange() {
    const target = getHashTarget();
    if (target) goToElement(target);
  }

  // add arrow to heading
  function addArrow(heading) {
    // add arrow button
    const arrow = document.createElement("button");
    arrow.innerHTML = document.querySelector(".icon_angle_down").innerHTML;
    arrow.classList.add("icon_button", "accordion_arrow");
    heading.insertBefore(arrow, heading.firstChild);

    // attach click listener to heading and button
    heading.addEventListener("click", onHeadingClick);
    arrow.addEventListener("click", onArrowClick);
  }

  // determine if on mobile-like device with small screen
  function isSmallScreen() {
    return Math.min(window.innerWidth, window.innerHeight) < 480;
  }

  // when <h2> heading is clicked
  function onHeadingClick(event) {
    // only collapse if <h2> itself is target of click (eg, user did
    // not click on anchor within <h2>)
    if (event.target === this) toggleCollapse(this);
  }

  // when arrow button is clicked
  function onArrowClick() {
    toggleCollapse(this.parentNode);
  }

  // collapse section if expanded, expand if collapsed
  function toggleCollapse(heading) {
    if (heading.dataset.collapsed === "false") collapseHeading(heading);
    else expandElement(heading);
  }

  // elements to exclude from collapse, such as table of contents panel,
  // hypothesis panel, etc
  const exclude = "#toc_panel, div.annotator-frame, #lightbox_overlay";

  // collapse section
  function collapseHeading(heading) {
    heading.setAttribute("data-collapsed", "true");
    const children = getChildren(heading);
    for (const child of children) child.setAttribute("data-collapsed", "true");
  }

  // expand section
  function expandElement(heading) {
    heading.setAttribute("data-collapsed", "false");
    const children = getChildren(heading);
    for (const child of children) child.setAttribute("data-collapsed", "false");
  }

  // get list of elements between this <h2> and next <h2> or <h1>
  // ("children" of the <h2> section)
  function getChildren(heading) {
    return nextUntil(heading, "h2, h1", exclude);
  }

  // start script when document is finished loading
  window.addEventListener("load", start);
</script>

<!-- angle down icon -->

<template class="icon_angle_down">
  <!-- modified from: https://fontawesome.com/icons/angle-down -->
  <svg width="16" height="16" viewBox="0 0 448 512">
    <path
      fill="currentColor"
      d="M207.029 381.476L12.686 187.132c-9.373-9.373-9.373-24.569 0-33.941l22.667-22.667c9.357-9.357 24.522-9.375 33.901-.04L224 284.505l154.745-154.021c9.379-9.335 24.544-9.317 33.901.04l22.667 22.667c9.373 9.373 9.373 24.569 0 33.941L240.971 381.476c-9.373 9.372-24.569 9.372-33.942 0z"
    ></path>
  </svg>
</template>

<style>
  @media only screen {
    /* accordion arrow button */
    .accordion_arrow {
      margin-right: 10px;
    }

    /* arrow icon when <h2> data-collapsed attribute true */
    h2[data-collapsed="true"] > .accordion_arrow > svg {
      transform: rotate(-90deg);
    }

    /* all elements (except <h2>'s) when data-collapsed attribute true */
    *:not(h2)[data-collapsed="true"] {
      display: none;
    }

    /* accordion arrow button when hovered and <h2>'s when hovered */
    .accordion_arrow:hover,
    h2[data-collapsed="true"]:hover,
    h2[data-collapsed="false"]:hover {
      cursor: pointer;
    }
  }

  /* always hide accordion arrow button on print */
  @media only print {
    .accordion_arrow {
      display: none;
    }
  }
</style>
<!--
  Anchors Plugin

  Adds an anchor next to each of a certain type of element that provides a
  human-readable url to that specific item/position in the document (e.g.
  "manuscript.html#abstract"). It also makes it such that scrolling out of view
  of a target removes its identifier from the url.
-->

<script type="module">
  // which types of elements to add anchors next to, in "document.querySelector"
  // format
  const typesQuery =
    'h1, h2, h3, div[id^="fig:"], div[id^="tbl:"], span[id^="eq:"]';

  // start script
  function start() {
    // add anchor to each element of specified types
    const elements = document.querySelectorAll(typesQuery);
    for (const element of elements) addAnchor(element);

    // attach scroll listener to window
    window.addEventListener("scroll", onScroll);
  }

  // when window is scrolled
  function onScroll() {
    // if url has hash and user has scrolled out of view of hash
    // target, remove hash from url
    const tolerance = 100;
    const target = getHashTarget();
    if (target) {
      if (
        target.getBoundingClientRect().top > window.innerHeight + tolerance ||
        target.getBoundingClientRect().bottom < 0 - tolerance
      )
        history.pushState(null, null, " ");
    }
  }

  // add anchor to element
  function addAnchor(element) {
    let addTo; // element to add anchor button to

    // if figure or table, modify withId and addTo to get expected
    // elements
    if (element.id.indexOf("fig:") === 0) {
      addTo = element.querySelector("figcaption");
    } else if (element.id.indexOf("tbl:") === 0) {
      addTo = element.querySelector("caption");
    } else if (element.id.indexOf("eq:") === 0) {
      addTo = element.querySelector(".eqnos-number");
    }

    addTo = addTo || element;
    const id = element.id || null;

    // do not add anchor if element doesn't have assigned id.
    // id is generated by pandoc and is assumed to be unique and
    // human-readable
    if (!id) return;

    // create anchor button
    const anchor = document.createElement("a");
    anchor.innerHTML = document.querySelector(".icon_link").innerHTML;
    anchor.title = "Link to this part of the document";
    anchor.classList.add("icon_button", "anchor");
    anchor.dataset.ignore = "true";
    anchor.href = "#" + id;
    addTo.appendChild(anchor);
  }

  // start script when document is finished loading
  window.addEventListener("load", start);
</script>

<!-- link icon -->

<template class="icon_link">
  <!-- modified from: https://fontawesome.com/icons/link -->
  <svg width="16" height="16" viewBox="0 0 512 512">
    <path
      fill="currentColor"
      d="M326.612 185.391c59.747 59.809 58.927 155.698.36 214.59-.11.12-.24.25-.36.37l-67.2 67.2c-59.27 59.27-155.699 59.262-214.96 0-59.27-59.26-59.27-155.7 0-214.96l37.106-37.106c9.84-9.84 26.786-3.3 27.294 10.606.648 17.722 3.826 35.527 9.69 52.721 1.986 5.822.567 12.262-3.783 16.612l-13.087 13.087c-28.026 28.026-28.905 73.66-1.155 101.96 28.024 28.579 74.086 28.749 102.325.51l67.2-67.19c28.191-28.191 28.073-73.757 0-101.83-3.701-3.694-7.429-6.564-10.341-8.569a16.037 16.037 0 0 1-6.947-12.606c-.396-10.567 3.348-21.456 11.698-29.806l21.054-21.055c5.521-5.521 14.182-6.199 20.584-1.731a152.482 152.482 0 0 1 20.522 17.197zM467.547 44.449c-59.261-59.262-155.69-59.27-214.96 0l-67.2 67.2c-.12.12-.25.25-.36.37-58.566 58.892-59.387 154.781.36 214.59a152.454 152.454 0 0 0 20.521 17.196c6.402 4.468 15.064 3.789 20.584-1.731l21.054-21.055c8.35-8.35 12.094-19.239 11.698-29.806a16.037 16.037 0 0 0-6.947-12.606c-2.912-2.005-6.64-4.875-10.341-8.569-28.073-28.073-28.191-73.639 0-101.83l67.2-67.19c28.239-28.239 74.3-28.069 102.325.51 27.75 28.3 26.872 73.934-1.155 101.96l-13.087 13.087c-4.35 4.35-5.769 10.79-3.783 16.612 5.864 17.194 9.042 34.999 9.69 52.721.509 13.906 17.454 20.446 27.294 10.606l37.106-37.106c59.271-59.259 59.271-155.699.001-214.959z"
    ></path>
  </svg>
</template>

<style>
  @media only screen {
    /* anchor button */
    .anchor {
      opacity: 0;
      margin-left: 5px;
    }

    /* anchor buttons within <h2>'s */
    h2 .anchor {
      margin-left: 10px;
    }

    /* anchor buttons when hovered/focused and anything containing an anchor button when hovered */
    *:hover > .anchor,
    .anchor:hover,
    .anchor:focus {
      opacity: 1;
    }

    /* anchor button when hovered */
    .anchor:hover {
      cursor: pointer;
    }
  }

  /* always show anchor button on devices with no mouse/hover ability */
  @media (hover: none) {
    .anchor {
      opacity: 1;
    }
  }

  /* always hide anchor button on print */
  @media only print {
    .anchor {
      display: none;
    }
  }
</style>
<!-- 
    Attributes Plugin

    Allows arbitrary HTML attributes to be attached to (almost) any element.
    Place an HTML comment inside or next to the desired element with the content:
    $attribute="value"
-->

<script type="module">
  // start script
  function start() {
    // get list of comments in document
    const comments = findComments();

    for (const comment of comments)
      if (comment.parentElement)
        addAttributes(comment.parentElement, comment.nodeValue.trim());
  }

  // add html attributes to specified element based on string of
  // html attributes and values
  function addAttributes(element, text) {
    // regex's for finding attribute/value pairs in the format of
    // attribute="value" or attribute='value
    const regex2 = /\$([a-zA-Z\-]+)?=\"(.+?)\"/;
    const regex1 = /\$([a-zA-Z\-]+)?=\'(.+?)\'/;

    // loop through attribute/value pairs
    let match;
    while ((match = text.match(regex2) || text.match(regex1))) {
      // get attribute and value from regex capture groups
      let attribute = match[1];
      let value = match[2];

      // remove from string
      text = text.substring(match.index + match[0].length);

      if (!attribute || !value) break;

      // set attribute of parent element
      try {
        element.setAttribute(attribute, value);
      } catch (error) {
        console.log(error);
      }

      // special case for colspan
      if (attribute === "colspan") removeTableCells(element, value);
    }
  }

  // get list of comment elements in document
  function findComments() {
    const comments = [];

    // iterate over comment nodes in document
    function acceptNode(node) {
      return NodeFilter.FILTER_ACCEPT;
    }
    const iterator = document.createNodeIterator(
      document.body,
      NodeFilter.SHOW_COMMENT,
      acceptNode
    );
    let node;
    while ((node = iterator.nextNode())) comments.push(node);

    return comments;
  }

  // remove certain number of cells after specified cell
  function removeTableCells(cell, number) {
    number = parseInt(number);
    if (!number) return;

    // remove elements
    for (; number > 1; number--) {
      if (cell.nextElementSibling) cell.nextElementSibling.remove();
    }
  }

  // start script on DOMContentLoaded instead of load to ensure this plugins
  // runs before other plugins
  window.addEventListener("DOMContentLoaded", start);
</script>
<!--
  Jump to First Plugin

  Adds a button next to each reference entry, figure, and table that jumps the
  page to the first occurrence of a link to that item in the manuscript.
-->

<script type="module">
  // whether to add buttons next to reference entries
  const references = "true";
  // whether to add buttons next to figures
  const figures = "true";
  // whether to add buttons next to tables
  const tables = "true";

  // start script
  function start() {
    if (references !== "false")
      makeButtons(`div[id^="ref-"]`, ".csl-left-margin", "reference");
    if (figures !== "false")
      makeButtons(`div[id^="fig:"]`, "figcaption", "figure");
    if (tables !== "false") makeButtons(`div[id^="tbl:"]`, "caption", "table");
  }

  // when jump button clicked
  function onButtonClick() {
    const first = getFirstOccurrence(this.dataset.id);
    if (!first) return;

    // update url hash so navigating "back" in history will return user to button
    window.location.hash = this.dataset.id;
    // scroll to link
    const timeout = function () {
      goToElement(first, window.innerHeight * 0.5);
    };
    window.setTimeout(timeout, 0);
  }

  // get first occurrence of link to item in document
  function getFirstOccurrence(id) {
    let query = "a";
    query += '[href="#' + id + '"]';
    // exclude buttons, anchor links, toc links, etc
    query += ":not(.button):not(.icon_button):not(.anchor):not(.toc_link)";
    return document.querySelector(query);
  }

  // add button next to each reference entry, figure, or table
  function makeButtons(query, containerQuery, subject) {
    const elements = document.querySelectorAll(query);
    for (const element of elements) {
      const id = element.id;
      const buttonContainer = element.querySelector(containerQuery);
      const first = getFirstOccurrence(id);

      // if can't find link to reference or place to put button, ignore
      if (!first || !buttonContainer) continue;

      // make jump button
      let button = document.createElement("button");
      button.classList.add("icon_button", "jump_arrow");
      button.title = `Jump to the first occurrence of this ${subject} in the document`;
      const icon = document.querySelector(".icon_angle_double_up");
      button.innerHTML = icon.innerHTML;
      button.dataset.id = id;
      button.dataset.ignore = "true";
      button.addEventListener("click", onButtonClick);
      buttonContainer.prepend(button);
    }
  }

  // start script when document is finished loading
  window.addEventListener("load", start);
</script>

<!-- angle double up icon -->

<template class="icon_angle_double_up">
  <!-- modified from: https://fontawesome.com/icons/angle-double-up -->
  <svg width="16" height="16" viewBox="0 0 320 512">
    <path
      fill="currentColor"
      d="M177 255.7l136 136c9.4 9.4 9.4 24.6 0 33.9l-22.6 22.6c-9.4 9.4-24.6 9.4-33.9 0L160 351.9l-96.4 96.4c-9.4 9.4-24.6 9.4-33.9 0L7 425.7c-9.4-9.4-9.4-24.6 0-33.9l136-136c9.4-9.5 24.6-9.5 34-.1zm-34-192L7 199.7c-9.4 9.4-9.4 24.6 0 33.9l22.6 22.6c9.4 9.4 24.6 9.4 33.9 0l96.4-96.4 96.4 96.4c9.4 9.4 24.6 9.4 33.9 0l22.6-22.6c9.4-9.4 9.4-24.6 0-33.9l-136-136c-9.2-9.4-24.4-9.4-33.8 0z"
    ></path>
  </svg>
</template>

<style>
  @media only screen {
    /* jump button */
    .jump_arrow {
      position: relative;
      top: 0.125em;
      margin-right: 5px;
    }
  }

  /* always hide jump button on print */
  @media only print {
    .jump_arrow {
      display: none;
    }
  }
</style>
<!-- 
    Lightbox Plugin

    Makes it such that when a user clicks on an image, the image fills the
    screen and the user can pan/drag/zoom the image and navigate between other
    images in the document.
-->

<script type="module">
  // list of possible zoom/scale factors
  const zooms =
    "0.1, 0.25, 0.333333, 0.5, 0.666666, 0.75, 1, 1.25, 1.5, 1.75, 2, 2.5, 3, 3.5, 4, 5, 6, 7, 8";
  // whether to fit image to view ('fit'), display at 100% and shrink if
  // necessary ('shrink'), or always display at 100% ('100')
  const defaultZoom = "fit";
  // whether to zoom in/out toward center of view ('true') or mouse ('false')
  const centerZoom = "false";

  // start script
  function start() {
    // run through each <img> element
    const imgs = document.querySelectorAll("figure > img");
    let count = 1;
    for (const img of imgs) {
      img.classList.add("lightbox_document_img");
      img.dataset.number = count;
      img.dataset.total = imgs.length;
      img.addEventListener("click", openLightbox);
      count++;
    }

    // attach mouse and key listeners to window
    window.addEventListener("mousemove", onWindowMouseMove);
    window.addEventListener("keyup", onKeyUp);
  }

  // when mouse is moved anywhere in window
  function onWindowMouseMove(event) {
    window.mouseX = event.clientX;
    window.mouseY = event.clientY;
  }

  // when key pressed
  function onKeyUp(event) {
    if (!event || !event.key) return;

    switch (event.key) {
      // trigger click of prev button
      case "ArrowLeft":
        const prevButton = document.getElementById("lightbox_prev_button");
        if (prevButton) prevButton.click();
        break;
      // trigger click of next button
      case "ArrowRight":
        const nextButton = document.getElementById("lightbox_next_button");
        if (nextButton) nextButton.click();
        break;
      // close on esc
      case "Escape":
        closeLightbox();
        break;
    }
  }

  // open lightbox
  function openLightbox() {
    const lightbox = makeLightbox(this);
    if (!lightbox) return;

    blurBody(lightbox);
    document.body.appendChild(lightbox);
  }

  // make lightbox
  function makeLightbox(img) {
    // delete lightbox if it exists, start fresh
    closeLightbox();

    // create screen overlay containing lightbox
    const overlay = document.createElement("div");
    overlay.id = "lightbox_overlay";

    // create image info boxes
    const numberInfo = document.createElement("div");
    const zoomInfo = document.createElement("div");
    numberInfo.id = "lightbox_number_info";
    zoomInfo.id = "lightbox_zoom_info";

    // create container for image
    const imageContainer = document.createElement("div");
    imageContainer.id = "lightbox_image_container";
    const lightboxImg = makeLightboxImg(
      img,
      imageContainer,
      numberInfo,
      zoomInfo
    );
    imageContainer.appendChild(lightboxImg);

    // create bottom container for caption and navigation buttons
    const bottomContainer = document.createElement("div");
    bottomContainer.id = "lightbox_bottom_container";
    const caption = makeCaption(img);
    const prevButton = makePrevButton(img);
    const nextButton = makeNextButton(img);
    bottomContainer.appendChild(prevButton);
    bottomContainer.appendChild(caption);
    bottomContainer.appendChild(nextButton);

    // attach top middle and bottom to overlay
    overlay.appendChild(numberInfo);
    overlay.appendChild(zoomInfo);
    overlay.appendChild(imageContainer);
    overlay.appendChild(bottomContainer);

    return overlay;
  }

  // make <img> object that is intuitively draggable and zoomable
  function makeLightboxImg(sourceImg, container, numberInfoBox, zoomInfoBox) {
    // create copy of source <img>
    const img = sourceImg.cloneNode(true);
    img.classList.remove("lightbox_document_img");
    img.removeAttribute("id");
    img.removeAttribute("width");
    img.removeAttribute("height");
    img.style.position = "unset";
    img.style.margin = "0";
    img.style.padding = "0";
    img.style.width = "";
    img.style.height = "";
    img.style.minWidth = "";
    img.style.minHeight = "";
    img.style.maxWidth = "";
    img.style.maxHeight = "";
    img.id = "lightbox_img";

    // build sorted list of zoomSteps
    const zoomSteps = zooms.split(/[^0-9.]/).map((step) => parseFloat(step));
    zoomSteps.sort((a, b) => a - b);

    // <img> object property variables
    let zoom = 1;
    let translateX = 0;
    let translateY = 0;
    let clickMouseX = undefined;
    let clickMouseY = undefined;
    let clickTranslateX = undefined;
    let clickTranslateY = undefined;

    updateNumberInfo();

    // update image numbers displayed in info box
    function updateNumberInfo() {
      numberInfoBox.innerHTML =
        sourceImg.dataset.number + " of " + sourceImg.dataset.total;
    }

    // update zoom displayed in info box
    function updateZoomInfo() {
      let zoomInfo = zoom * 100;
      if (!Number.isInteger(zoomInfo)) zoomInfo = zoomInfo.toFixed(2);
      zoomInfoBox.innerHTML = zoomInfo + "%";
    }

    // move to closest zoom step above current zoom
    const zoomIn = function () {
      for (const zoomStep of zoomSteps) {
        if (zoomStep > zoom) {
          zoom = zoomStep;
          break;
        }
      }
      updateTransform();
    };

    // move to closest zoom step above current zoom
    const zoomOut = function () {
      zoomSteps.reverse();
      for (const zoomStep of zoomSteps) {
        if (zoomStep < zoom) {
          zoom = zoomStep;
          break;
        }
      }
      zoomSteps.reverse();

      updateTransform();
    };

    // update display of <img> based on scale/translate properties
    const updateTransform = function () {
      // set transform
      img.style.transform =
        "translate(" +
        (translateX || 0) +
        "px," +
        (translateY || 0) +
        "px) scale(" +
        (zoom || 1) +
        ")";

      // get new width/height after scale
      const rect = img.getBoundingClientRect();
      // limit translate
      translateX = Math.max(translateX, -rect.width / 2);
      translateX = Math.min(translateX, rect.width / 2);
      translateY = Math.max(translateY, -rect.height / 2);
      translateY = Math.min(translateY, rect.height / 2);

      // set transform
      img.style.transform =
        "translate(" +
        (translateX || 0) +
        "px," +
        (translateY || 0) +
        "px) scale(" +
        (zoom || 1) +
        ")";

      updateZoomInfo();
    };

    // fit <img> to container
    const fit = function () {
      // no x/y offset, 100% zoom by default
      translateX = 0;
      translateY = 0;
      zoom = 1;

      // widths of <img> and container
      const imgWidth = img.naturalWidth;
      const imgHeight = img.naturalHeight;
      const containerWidth = parseFloat(
        window.getComputedStyle(container).width
      );
      const containerHeight = parseFloat(
        window.getComputedStyle(container).height
      );

      // how much zooming is needed to fit <img> to container
      const xRatio = imgWidth / containerWidth;
      const yRatio = imgHeight / containerHeight;
      const maxRatio = Math.max(xRatio, yRatio);
      const newZoom = 1 / maxRatio;

      // fit <img> to container according to option
      if (defaultZoom === "shrink") {
        if (maxRatio > 1) zoom = newZoom;
      } else if (defaultZoom === "fit") zoom = newZoom;

      updateTransform();
    };

    // when mouse wheel is rolled anywhere in container
    const onContainerWheel = function (event) {
      if (!event) return;

      // let ctrl + mouse wheel to zoom behave as normal
      if (event.ctrlKey) return;

      // prevent normal scroll behavior
      event.preventDefault();
      event.stopPropagation();

      // point around which to scale img
      const viewRect = container.getBoundingClientRect();
      const viewX = (viewRect.left + viewRect.right) / 2;
      const viewY = (viewRect.top + viewRect.bottom) / 2;
      const originX = centerZoom === "true" ? viewX : mouseX;
      const originY = centerZoom === "true" ? viewY : mouseY;

      // get point on image under origin
      const oldRect = img.getBoundingClientRect();
      const oldPercentX = (originX - oldRect.left) / oldRect.width;
      const oldPercentY = (originY - oldRect.top) / oldRect.height;

      // increment/decrement zoom
      if (event.deltaY < 0) zoomIn();
      if (event.deltaY > 0) zoomOut();

      // get offset between previous image point and origin
      const newRect = img.getBoundingClientRect();
      const offsetX = originX - (newRect.left + newRect.width * oldPercentX);
      const offsetY = originY - (newRect.top + newRect.height * oldPercentY);

      // translate image to keep image point under origin
      translateX += offsetX;
      translateY += offsetY;

      // perform translate
      updateTransform();
    };

    // when container is clicked
    function onContainerClick(event) {
      // if container itself is target of click, and not other
      // element above it
      if (event.target === this) closeLightbox();
    }

    // when mouse button is pressed on image
    const onImageMouseDown = function (event) {
      // store original mouse position relative to image
      clickMouseX = window.mouseX;
      clickMouseY = window.mouseY;
      clickTranslateX = translateX;
      clickTranslateY = translateY;
      event.stopPropagation();
      event.preventDefault();
    };

    // when mouse button is released anywhere in window
    const onWindowMouseUp = function (event) {
      // reset original mouse position
      clickMouseX = undefined;
      clickMouseY = undefined;
      clickTranslateX = undefined;
      clickTranslateY = undefined;

      // remove global listener if lightbox removed from document
      if (!document.body.contains(container))
        window.removeEventListener("mouseup", onWindowMouseUp);
    };

    // when mouse is moved anywhere in window
    const onWindowMouseMove = function (event) {
      if (
        clickMouseX === undefined ||
        clickMouseY === undefined ||
        clickTranslateX === undefined ||
        clickTranslateY === undefined
      )
        return;

      // offset image based on original and current mouse position
      translateX = clickTranslateX + window.mouseX - clickMouseX;
      translateY = clickTranslateY + window.mouseY - clickMouseY;
      updateTransform();
      event.preventDefault();

      // remove global listener if lightbox removed from document
      if (!document.body.contains(container))
        window.removeEventListener("mousemove", onWindowMouseMove);
    };

    // when window is resized
    const onWindowResize = function (event) {
      fit();

      // remove global listener if lightbox removed from document
      if (!document.body.contains(container))
        window.removeEventListener("resize", onWindowResize);
    };

    // attach the necessary event listeners
    img.addEventListener("dblclick", fit);
    img.addEventListener("mousedown", onImageMouseDown);
    container.addEventListener("wheel", onContainerWheel);
    container.addEventListener("mousedown", onContainerClick);
    container.addEventListener("touchstart", onContainerClick);
    window.addEventListener("mouseup", onWindowMouseUp);
    window.addEventListener("mousemove", onWindowMouseMove);
    window.addEventListener("resize", onWindowResize);

    // run fit() after lightbox atttached to document and <img> Loaded
    // so needed container and img dimensions available
    img.addEventListener("load", fit);

    return img;
  }

  // make caption
  function makeCaption(img) {
    const caption = document.createElement("div");
    caption.id = "lightbox_caption";
    const captionSource = img.nextElementSibling;
    if (captionSource.tagName.toLowerCase() === "figcaption") {
      const captionCopy = makeCopy(captionSource);
      caption.innerHTML = captionCopy.innerHTML;
    }

    caption.addEventListener("touchstart", function (event) {
      event.stopPropagation();
    });

    return caption;
  }

  // make carbon copy of html dom element
  function makeCopy(source) {
    const sourceCopy = source.cloneNode(true);

    // delete elements marked with ignore (eg anchor and jump buttons)
    const deleteFromCopy = sourceCopy.querySelectorAll('[data-ignore="true"]');
    for (const element of deleteFromCopy) element.remove();

    // delete certain element attributes
    const attributes = [
      "id",
      "data-collapsed",
      "data-selected",
      "data-highlighted",
      "data-glow",
    ];
    for (const attribute of attributes) {
      sourceCopy.removeAttribute(attribute);
      const elements = sourceCopy.querySelectorAll("[" + attribute + "]");
      for (const element of elements) element.removeAttribute(attribute);
    }

    return sourceCopy;
  }

  // make button to jump to previous image in document
  function makePrevButton(img) {
    const prevButton = document.createElement("button");
    prevButton.id = "lightbox_prev_button";
    prevButton.title = "Jump to the previous image in the document [←]";
    prevButton.classList.add("icon_button", "lightbox_button");
    prevButton.innerHTML = document.querySelector(".icon_caret_left").innerHTML;

    // attach click listeners to button
    prevButton.addEventListener("click", function () {
      getPrevImg(img).click();
    });

    return prevButton;
  }

  // make button to jump to next image in document
  function makeNextButton(img) {
    const nextButton = document.createElement("button");
    nextButton.id = "lightbox_next_button";
    nextButton.title = "Jump to the next image in the document [→]";
    nextButton.classList.add("icon_button", "lightbox_button");
    nextButton.innerHTML = document.querySelector(
      ".icon_caret_right"
    ).innerHTML;

    // attach click listeners to button
    nextButton.addEventListener("click", function () {
      getNextImg(img).click();
    });

    return nextButton;
  }

  // get previous image in document
  function getPrevImg(img) {
    const imgs = document.querySelectorAll(".lightbox_document_img");

    // find index of provided img
    let index;
    for (index = 0; index < imgs.length; index++) {
      if (imgs[index] === img) break;
    }

    // wrap index to other side if < 1
    if (index - 1 >= 0) index--;
    else index = imgs.length - 1;
    return imgs[index];
  }

  // get next image in document
  function getNextImg(img) {
    const imgs = document.querySelectorAll(".lightbox_document_img");

    // find index of provided img
    let index;
    for (index = 0; index < imgs.length; index++) {
      if (imgs[index] === img) break;
    }

    // wrap index to other side if > total
    if (index + 1 <= imgs.length - 1) index++;
    else index = 0;
    return imgs[index];
  }

  // close lightbox
  function closeLightbox() {
    focusBody();

    const lightbox = document.getElementById("lightbox_overlay");
    if (lightbox) lightbox.remove();
  }

  // make all elements behind lightbox non-focusable
  function blurBody(overlay) {
    const all = document.querySelectorAll("*");
    for (const element of all) element.tabIndex = -1;
    document.body.classList.add("body_no_scroll");
  }

  // make all elements focusable again
  function focusBody() {
    const all = document.querySelectorAll("*");
    for (const element of all) element.removeAttribute("tabIndex");
    document.body.classList.remove("body_no_scroll");
  }

  // start script when document is finished loading
  window.addEventListener("load", start);
</script>

<!-- caret left icon -->

<template class="icon_caret_left">
  <!-- modified from: https://fontawesome.com/icons/caret-left -->
  <svg width="16" height="16" viewBox="0 0 192 512">
    <path
      fill="currentColor"
      d="M192 127.338v257.324c0 17.818-21.543 26.741-34.142 14.142L29.196 270.142c-7.81-7.81-7.81-20.474 0-28.284l128.662-128.662c12.599-12.6 34.142-3.676 34.142 14.142z"
    ></path>
  </svg>
</template>

<!-- caret right icon -->

<template class="icon_caret_right">
  <!-- modified from: https://fontawesome.com/icons/caret-right -->
  <svg width="16" height="16" viewBox="0 0 192 512">
    <path
      fill="currentColor"
      d="M0 384.662V127.338c0-17.818 21.543-26.741 34.142-14.142l128.662 128.662c7.81 7.81 7.81 20.474 0 28.284L34.142 398.804C21.543 411.404 0 402.48 0 384.662z"
    ></path>
  </svg>
</template>

<style>
  @media only screen {
    /* regular <img> in document when hovered */
    img.lightbox_document_img:hover {
      cursor: pointer;
    }

    .body_no_scroll {
      overflow: hidden !important;
    }

    /* screen overlay */
    #lightbox_overlay {
      display: flex;
      flex-direction: column;
      position: fixed;
      left: 0;
      top: 0;
      right: 0;
      bottom: 0;
      background: rgba(0, 0, 0, 0.75);
      z-index: 3;
    }

    /* middle area containing lightbox image */
    #lightbox_image_container {
      flex-grow: 1;
      display: flex;
      justify-content: center;
      align-items: center;
      overflow: hidden;
      position: relative;
      padding: 20px;
    }

    /* bottom area containing caption */
    #lightbox_bottom_container {
      display: flex;
      justify-content: center;
      align-items: center;
      height: 100px;
      min-height: 100px;
      max-height: 100px;
      background: rgba(0, 0, 0, 0.5);
    }

    /* image number info text box */
    #lightbox_number_info {
      position: absolute;
      color: #ffffff;
      font-weight: 600;
      left: 2px;
      top: 0;
      z-index: 4;
    }

    /* zoom info text box */
    #lightbox_zoom_info {
      position: absolute;
      color: #ffffff;
      font-weight: 600;
      right: 2px;
      top: 0;
      z-index: 4;
    }

    /* copy of image caption */
    #lightbox_caption {
      box-sizing: border-box;
      display: inline-block;
      width: 100%;
      max-height: 100%;
      padding: 10px 0;
      text-align: center;
      overflow-y: auto;
      color: #ffffff;
    }

    /* navigation previous/next button */
    .lightbox_button {
      width: 100px;
      height: 100%;
      min-width: 100px;
      min-height: 100%;
      color: #ffffff;
    }

    /* navigation previous/next button when hovered */
    .lightbox_button:hover {
      background: none !important;
    }

    /* navigation button icon */
    .lightbox_button > svg {
      height: 25px;
    }

    /* figure auto-number */
    #lightbox_caption > span:first-of-type {
      font-weight: bold;
      margin-right: 5px;
    }

    /* lightbox image when hovered */
    #lightbox_img:hover {
      cursor: grab;
    }

    /* lightbox image when grabbed */
    #lightbox_img:active {
      cursor: grabbing;
    }
  }

  /* when on screen < 480px wide */
  @media only screen and (max-width: 480px) {
    /* make navigation buttons skinnier on small screens to make more room for caption text */
    .lightbox_button {
      width: 50px;
      min-width: 50px;
    }
  }

  /* always hide lightbox on print */
  @media only print {
    #lightbox_overlay {
      display: none;
    }
  }
</style>
<!-- 
  Link Highlight Plugin

  Makes it such that when a user hovers or focuses a link, other links that have
  the same target will be highlighted. It also makes it such that when clicking
  a link, the target of the link (eg reference, figure, table) is briefly
  highlighted.
-->

<script type="module">
  // whether to also highlight links that go to external urls
  const externalLinks = "false";
  // whether user must click off to unhighlight instead of just
  // un-hovering
  const clickUnhighlight = "false";
  // whether to also highlight links that are unique
  const highlightUnique = "true";

  // start script
  function start() {
    const links = getLinks();
    for (const link of links) {
      // attach mouse and focus listeners to link
      link.addEventListener("mouseenter", onLinkFocus);
      link.addEventListener("focus", onLinkFocus);
      link.addEventListener("mouseleave", onLinkUnhover);
    }

    // attach click and hash change listeners to window
    window.addEventListener("click", onClick);
    window.addEventListener("touchstart", onClick);
    window.addEventListener("hashchange", onHashChange);

    // run hash change on window load in case user has navigated
    // directly to hash
    onHashChange();
  }

  // when link is focused (tabbed to) or hovered
  function onLinkFocus() {
    highlight(this);
  }

  // when link is unhovered
  function onLinkUnhover() {
    if (clickUnhighlight !== "true") unhighlightAll();
  }

  // when the mouse is clicked anywhere in window
  function onClick(event) {
    unhighlightAll();
  }

  // when hash (eg manuscript.html#introduction) changes
  function onHashChange() {
    const target = getHashTarget();
    if (target) glowElement(target);
  }

  // start glow sequence on an element
  function glowElement(element) {
    const startGlow = function () {
      onGlowEnd();
      element.dataset.glow = "true";
      element.addEventListener("animationend", onGlowEnd);
    };
    const onGlowEnd = function () {
      element.removeAttribute("data-glow");
      element.removeEventListener("animationend", onGlowEnd);
    };
    startGlow();
  }

  // highlight link and all others with same target
  function highlight(link) {
    // force unhighlight all to start fresh
    unhighlightAll();

    // get links with same target
    if (!link) return;
    const sameLinks = getSameLinks(link);

    // if link unique and option is off, exit and don't highlight
    if (sameLinks.length <= 1 && highlightUnique !== "true") return;

    // highlight all same links, and "select" (special highlight) this
    // one
    for (const sameLink of sameLinks) {
      if (sameLink === link) sameLink.setAttribute("data-selected", "true");
      else sameLink.setAttribute("data-highlighted", "true");
    }
  }

  // unhighlight all links
  function unhighlightAll() {
    const links = getLinks();
    for (const link of links) {
      link.setAttribute("data-selected", "false");
      link.setAttribute("data-highlighted", "false");
    }
  }

  // get links with same target
  function getSameLinks(link) {
    const results = [];
    const links = getLinks();
    for (const otherLink of links) {
      if (otherLink.getAttribute("href") === link.getAttribute("href"))
        results.push(otherLink);
    }
    return results;
  }

  // get all links of types we wish to handle
  function getLinks() {
    let query = "a";
    if (externalLinks !== "true") query += '[href^="#"]';
    // exclude buttons, anchor links, toc links, etc
    query += ":not(.button):not(.icon_button):not(.anchor):not(.toc_link)";
    return document.querySelectorAll(query);
  }

  // start script when document is finished loading
  window.addEventListener("load", start);
</script>

<style>
  @media only screen {
    /* anything with data-highlighted attribute true */
    [data-highlighted="true"] {
      background: #ffeb3b;
    }

    /* anything with data-selected attribute true */
    [data-selected="true"] {
      background: #ff8a65 !important;
    }

    /* animation definition for glow */
    @keyframes highlight_glow {
      0% {
        background: none;
      }
      10% {
        background: #bbdefb;
      }
      100% {
        background: none;
      }
    }

    /* anything with data-glow attribute true */
    [data-glow="true"] {
      animation: highlight_glow 2s;
    }
  }
</style>
<!--
  Table of Contents Plugin

  Provides a "table of contents" (toc) panel on the side of the document that
  allows the user to conveniently navigate between sections of the document.
-->

<script type="module">
  // which types of elements to add links for, in "document.querySelector" format
  const typesQuery = "h1, h2, h3";
  // whether toc starts open. use 'true' or 'false', or 'auto' to
  // use 'true' behavior when screen wide enough and 'false' when not
  const startOpen = "false";
  // whether toc closes when clicking on toc link. use 'true' or
  // 'false', or 'auto' to use 'false' behavior when screen wide
  // enough and 'true' when not
  const clickClose = "auto";
  // if list item is more than this many characters, text will be
  // truncated
  const charLimit = "50";
  // whether or not to show bullets next to each toc item
  const bullets = "false";

  // start script
  function start() {
    // make toc panel and populate with entries (links to document
    // sections)
    const panel = makePanel();
    if (!panel) return;
    makeEntries(panel);
    // attach panel to document after making entries, so 'toc' heading
    // in panel isn't included in toc
    document.body.insertBefore(panel, document.body.firstChild);

    // initial panel state
    if (startOpen === "true" || (startOpen === "auto" && !isSmallScreen()))
      openPanel();
    else closePanel();

    // attach click, scroll, and hash change listeners to window
    window.addEventListener("click", onClick);
    window.addEventListener("scroll", onScroll);
    window.addEventListener("hashchange", onScroll);
    window.addEventListener("keyup", onKeyUp);
    onScroll();

    // add class to push document body down out of way of toc button
    document.body.classList.add("toc_body_nudge");
  }

  // determine if screen wide enough to fit toc panel
  function isSmallScreen() {
    // in default theme:
    // 816px = 8.5in = width of "page" (<body>) element
    // 260px = min width of toc panel (*2 for both sides of <body>)
    return window.innerWidth < 816 + 260 * 2;
  }

  // when mouse is clicked anywhere in window
  function onClick() {
    if (isSmallScreen()) closePanel();
  }

  // when window is scrolled or hash changed
  function onScroll() {
    highlightViewed();
  }

  // when key pressed
  function onKeyUp(event) {
    if (!event || !event.key) return;

    // close on esc
    if (event.key === "Escape") closePanel();
  }

  // find entry of currently viewed document section in toc and highlight
  function highlightViewed() {
    const firstId = getFirstInView(typesQuery);

    // get toc entries (links), unhighlight all, then highlight viewed
    const list = document.getElementById("toc_list");
    if (!firstId || !list) return;
    const links = list.querySelectorAll("a");
    for (const link of links) link.dataset.viewing = "false";
    const link = list.querySelector('a[href="#' + firstId + '"]');
    if (!link) return;
    link.dataset.viewing = "true";
  }

  // get first or previous toc listed element in top half of view
  function getFirstInView(query) {
    // get all elements matching query and with id
    const elements = document.querySelectorAll(query);
    const elementsWithIds = [];
    for (const element of elements) {
      if (element.id) elementsWithIds.push(element);
    }

    // get first or previous element in top half of view
    for (let i = 0; i < elementsWithIds.length; i++) {
      const element = elementsWithIds[i];
      const prevElement = elementsWithIds[Math.max(0, i - 1)];
      if (element.getBoundingClientRect().top >= 0) {
        if (element.getBoundingClientRect().top < window.innerHeight / 2)
          return element.id;
        else return prevElement.id;
      }
    }
  }

  // make panel
  function makePanel() {
    // create panel
    const panel = document.createElement("div");
    panel.id = "toc_panel";
    if (bullets === "true") panel.dataset.bullets = "true";

    // create header
    const header = document.createElement("div");
    header.id = "toc_header";

    // create toc button
    const button = document.createElement("button");
    button.id = "toc_button";
    button.innerHTML = document.querySelector(".icon_th_list").innerHTML;
    button.title = "Table of Contents";
    button.classList.add("icon_button");

    // create header text
    const text = document.createElement("h4");
    text.innerHTML = "Table of Contents";

    // create container for toc list
    const list = document.createElement("div");
    list.id = "toc_list";

    // attach click listeners
    panel.addEventListener("click", onPanelClick);
    header.addEventListener("click", onHeaderClick);
    button.addEventListener("click", onButtonClick);

    // attach elements
    header.appendChild(button);
    header.appendChild(text);
    panel.appendChild(header);
    panel.appendChild(list);

    return panel;
  }

  // create toc entries (links) to each element of the specified types
  function makeEntries(panel) {
    const elements = document.querySelectorAll(typesQuery);
    for (const element of elements) {
      // do not add link if element doesn't have assigned id
      if (!element.id) continue;

      // create link/list item
      const link = document.createElement("a");
      link.classList.add("toc_link");
      switch (element.tagName.toLowerCase()) {
        case "h1":
          link.dataset.level = "1";
          break;
        case "h2":
          link.dataset.level = "2";
          break;
        case "h3":
          link.dataset.level = "3";
          break;
        case "h4":
          link.dataset.level = "4";
          break;
      }
      link.title = element.innerText;
      let text = element.innerText;
      if (text.length > charLimit) text = text.slice(0, charLimit) + "...";
      link.innerHTML = text;
      link.href = "#" + element.id;
      link.addEventListener("click", onLinkClick);

      // attach link
      panel.querySelector("#toc_list").appendChild(link);
    }
  }

  // when panel is clicked
  function onPanelClick(event) {
    // stop click from propagating to window/document and closing panel
    event.stopPropagation();
  }

  // when header itself is clicked
  function onHeaderClick(event) {
    togglePanel();
  }

  // when button is clicked
  function onButtonClick(event) {
    togglePanel();
    // stop header underneath button from also being clicked
    event.stopPropagation();
  }

  // when link is clicked
  function onLinkClick(event) {
    if (clickClose === "true" || (clickClose === "auto" && isSmallScreen()))
      closePanel();
    else openPanel();
  }

  // open panel if closed, close if opened
  function togglePanel() {
    const panel = document.getElementById("toc_panel");
    if (!panel) return;

    if (panel.dataset.open === "true") closePanel();
    else openPanel();
  }

  // open panel
  function openPanel() {
    const panel = document.getElementById("toc_panel");
    if (panel) panel.dataset.open = "true";
  }

  // close panel
  function closePanel() {
    const panel = document.getElementById("toc_panel");
    if (panel) panel.dataset.open = "false";
  }

  // start script when document is finished loading
  window.addEventListener("load", start);
</script>

<!-- th list icon -->

<template class="icon_th_list">
  <!-- modified from: https://fontawesome.com/icons/th-list -->
  <svg width="16" height="16" viewBox="0 0 512 512" tabindex="-1">
    <path
      fill="currentColor"
      d="M96 96c0 26.51-21.49 48-48 48S0 122.51 0 96s21.49-48 48-48 48 21.49 48 48zM48 208c-26.51 0-48 21.49-48 48s21.49 48 48 48 48-21.49 48-48-21.49-48-48-48zm0 160c-26.51 0-48 21.49-48 48s21.49 48 48 48 48-21.49 48-48-21.49-48-48-48zm96-236h352c8.837 0 16-7.163 16-16V76c0-8.837-7.163-16-16-16H144c-8.837 0-16 7.163-16 16v40c0 8.837 7.163 16 16 16zm0 160h352c8.837 0 16-7.163 16-16v-40c0-8.837-7.163-16-16-16H144c-8.837 0-16 7.163-16 16v40c0 8.837 7.163 16 16 16zm0 160h352c8.837 0 16-7.163 16-16v-40c0-8.837-7.163-16-16-16H144c-8.837 0-16 7.163-16 16v40c0 8.837 7.163 16 16 16z"
      tabindex="-1"
    ></path>
  </svg>
</template>

<style>
  @media only screen {
    /* toc panel */
    #toc_panel {
      box-sizing: border-box;
      position: fixed;
      top: 0;
      left: 0;
      background: #ffffff;
      box-shadow: 0 0 20px rgba(0, 0, 0, 0.05);
      z-index: 2;
    }

    /* toc panel when closed */
    #toc_panel[data-open="false"] {
      min-width: 60px;
      width: 60px;
      height: 60px;
      border-right: solid 1px #bdbdbd;
      border-bottom: solid 1px #bdbdbd;
    }

    /* toc panel when open */
    #toc_panel[data-open="true"] {
      min-width: 260px;
      max-width: 480px;
      /* keep panel edge consistent distance away from "page" edge */
      width: calc(((100vw - 8.5in) / 2) - 30px - 40px);
      bottom: 0;
      border-right: solid 1px #bdbdbd;
    }

    /* toc panel header */
    #toc_header {
      box-sizing: border-box;
      display: flex;
      flex-direction: row;
      align-items: center;
      height: 60px;
      margin: 0;
      padding: 20px;
    }

    /* toc panel header when hovered */
    #toc_header:hover {
      cursor: pointer;
    }

    /* toc panel header when panel open */
    #toc_panel[data-open="true"] > #toc_header {
      border-bottom: solid 1px #bdbdbd;
    }

    /* toc open/close header button */
    #toc_button {
      margin-right: 20px;
    }

    /* hide toc list and header text when closed */
    #toc_panel[data-open="false"] > #toc_header > *:not(#toc_button),
    #toc_panel[data-open="false"] > #toc_list {
      display: none;
    }

    /* toc list of entries */
    #toc_list {
      box-sizing: border-box;
      width: 100%;
      padding: 20px;
      position: absolute;
      top: calc(60px + 1px);
      bottom: 0;
      overflow: auto;
    }

    /* toc entry, link to section in document */
    .toc_link {
      display: block;
      padding: 5px;
      position: relative;
      font-weight: 600;
      text-decoration: none;
    }

    /* toc entry when hovered or when "viewed" */
    .toc_link:hover,
    .toc_link[data-viewing="true"] {
      background: #f5f5f5;
    }

    /* toc entry, level 1 indentation */
    .toc_link[data-level="1"] {
      margin-left: 0;
    }

    /* toc entry, level 2 indentation */
    .toc_link[data-level="2"] {
      margin-left: 20px;
    }

    /* toc entry, level 3 indentation */
    .toc_link[data-level="3"] {
      margin-left: 40px;
    }

    /* toc entry, level 4 indentation */
    .toc_link[data-level="4"] {
      margin-left: 60px;
    }

    /* toc entry bullets */
    #toc_panel[data-bullets="true"] .toc_link[data-level]:before {
      position: absolute;
      left: -15px;
      top: -1px;
      font-size: 1.5em;
    }

    /* toc entry, level 2 bullet */
    #toc_panel[data-bullets="true"] .toc_link[data-level="2"]:before {
      content: "\2022";
    }

    /* toc entry, level 3 bullet */
    #toc_panel[data-bullets="true"] .toc_link[data-level="3"]:before {
      content: "\25AB";
    }

    /* toc entry, level 4 bullet */
    #toc_panel[data-bullets="true"] .toc_link[data-level="4"]:before {
      content: "-";
    }
  }

  /* when on screen < 8.5in wide */
  @media only screen and (max-width: 8.5in) {
    /* push <body> ("page") element down to make room for toc icon */
    .toc_body_nudge {
      padding-top: 60px;
    }

    /* toc icon when panel closed and not hovered */
    #toc_panel[data-open="false"]:not(:hover) {
      background: rgba(255, 255, 255, 0.75);
    }
  }

  /* always hide toc panel on print */
  @media only print {
    #toc_panel {
      display: none;
    }
  }
</style>
<!-- 
  Tooltips Plugin

  Makes it such that when the user hovers or focuses a link to a citation or
  figure, a tooltip appears with a preview of the reference content, along with
  arrows to navigate between instances of the same reference in the document.
-->

<script type="module">
  // whether user must click off to close tooltip instead of just un-hovering
  const clickClose = "false";
  // delay (in ms) between opening and closing tooltip
  const delay = "100";

  // start script
  function start() {
    const links = getLinks();
    for (const link of links) {
      // attach hover and focus listeners to link
      link.addEventListener("mouseover", onLinkHover);
      link.addEventListener("mouseleave", onLinkUnhover);
      link.addEventListener("focus", onLinkFocus);
      link.addEventListener("touchend", onLinkTouch);
    }

    // attach mouse, key, and resize listeners to window
    window.addEventListener("mousedown", onClick);
    window.addEventListener("touchstart", onClick);
    window.addEventListener("keyup", onKeyUp);
    window.addEventListener("resize", onResize);
  }

  // when link is hovered
  function onLinkHover() {
    // function to open tooltip
    const delayOpenTooltip = function () {
      openTooltip(this);
    }.bind(this);

    // run open function after delay
    this.openTooltipTimer = window.setTimeout(delayOpenTooltip, delay);
  }

  // when mouse leaves link
  function onLinkUnhover() {
    // cancel opening tooltip
    window.clearTimeout(this.openTooltipTimer);

    // don't close on unhover if option specifies
    if (clickClose === "true") return;

    // function to close tooltip
    const delayCloseTooltip = function () {
      // if tooltip open and if mouse isn't over tooltip, close
      const tooltip = document.getElementById("tooltip");
      if (tooltip && !tooltip.matches(":hover")) closeTooltip();
    };

    // run close function after delay
    this.closeTooltipTimer = window.setTimeout(delayCloseTooltip, delay);
  }

  // when link is focused (tabbed to)
  function onLinkFocus(event) {
    openTooltip(this);
  }

  // when link is touched on touch screen
  function onLinkTouch(event) {
    // attempt to force hover state on first tap always, and trigger
    // regular link click (and navigation) on second tap
    if (event.target === document.activeElement) event.target.click();
    else {
      document.activeElement.blur();
      event.target.focus();
    }
    if (event.cancelable) event.preventDefault();
    event.stopPropagation();
    return false;
  }

  // when mouse is clicked anywhere in window
  function onClick(event) {
    closeTooltip();
  }

  // when key pressed
  function onKeyUp(event) {
    if (!event || !event.key) return;

    switch (event.key) {
      // trigger click of prev button
      case "ArrowLeft":
        const prevButton = document.getElementById("tooltip_prev_button");
        if (prevButton) prevButton.click();
        break;
      // trigger click of next button
      case "ArrowRight":
        const nextButton = document.getElementById("tooltip_next_button");
        if (nextButton) nextButton.click();
        break;
      // close on esc
      case "Escape":
        closeTooltip();
        break;
    }
  }

  // when window is resized or zoomed
  function onResize() {
    closeTooltip();
  }

  // get all links of types we wish to handle
  function getLinks() {
    const queries = [];
    // exclude buttons, anchor links, toc links, etc
    const exclude =
      ":not(.button):not(.icon_button):not(.anchor):not(.toc_link)";
    queries.push('a[href^="#ref-"]' + exclude); // citation links
    queries.push('a[href^="#fig:"]' + exclude); // figure links
    const query = queries.join(", ");
    return document.querySelectorAll(query);
  }

  // get links with same target, get index of link in set, get total
  // same links
  function getSameLinks(link) {
    const sameLinks = [];
    const links = getLinks();
    for (const otherLink of links) {
      if (otherLink.getAttribute("href") === link.getAttribute("href"))
        sameLinks.push(otherLink);
    }

    return {
      elements: sameLinks,
      index: sameLinks.indexOf(link),
      total: sameLinks.length,
    };
  }

  // open tooltip
  function openTooltip(link) {
    // delete tooltip if it exists, start fresh
    closeTooltip();

    // make tooltip element
    const tooltip = makeTooltip(link);

    // if source couldn't be found and tooltip not made, exit
    if (!tooltip) return;

    // make navbar elements
    const navBar = makeNavBar(link);
    if (navBar) tooltip.firstElementChild.appendChild(navBar);

    // attach tooltip to page
    document.body.appendChild(tooltip);

    // position tooltip
    const position = function () {
      positionTooltip(link);
    };
    position();

    // if tooltip contains images, position again after they've loaded
    const imgs = tooltip.querySelectorAll("img");
    for (const img of imgs) img.addEventListener("load", position);
  }

  // close (delete) tooltip
  function closeTooltip() {
    const tooltip = document.getElementById("tooltip");
    if (tooltip) tooltip.remove();
  }

  // make tooltip
  function makeTooltip(link) {
    // get target element that link points to
    const source = getSource(link);

    // if source can't be found, exit
    if (!source) return;

    // create new tooltip
    const tooltip = document.createElement("div");
    tooltip.id = "tooltip";
    const tooltipContent = document.createElement("div");
    tooltipContent.id = "tooltip_content";
    tooltip.appendChild(tooltipContent);

    // make copy of source node and put in tooltip
    const sourceCopy = makeCopy(source);
    tooltipContent.appendChild(sourceCopy);

    // attach mouse event listeners
    tooltip.addEventListener("click", onTooltipClick);
    tooltip.addEventListener("mousedown", onTooltipClick);
    tooltip.addEventListener("touchstart", onTooltipClick);
    tooltip.addEventListener("mouseleave", onTooltipUnhover);

    // (for interaction with lightbox plugin)
    // transfer click on tooltip copied img to original img
    const sourceImg = source.querySelector("img");
    const sourceCopyImg = sourceCopy.querySelector("img");
    if (sourceImg && sourceCopyImg) {
      const clickImg = function () {
        sourceImg.click();
        closeTooltip();
      };
      sourceCopyImg.addEventListener("click", clickImg);
    }

    return tooltip;
  }

  // make carbon copy of html dom element
  function makeCopy(source) {
    const sourceCopy = source.cloneNode(true);

    // delete elements marked with ignore (eg anchor and jump buttons)
    const deleteFromCopy = sourceCopy.querySelectorAll('[data-ignore="true"]');
    for (const element of deleteFromCopy) element.remove();

    // delete certain element attributes
    const attributes = [
      "id",
      "data-collapsed",
      "data-selected",
      "data-highlighted",
      "data-glow",
      "class",
    ];
    for (const attribute of attributes) {
      sourceCopy.removeAttribute(attribute);
      const elements = sourceCopy.querySelectorAll("[" + attribute + "]");
      for (const element of elements) element.removeAttribute(attribute);
    }

    return sourceCopy;
  }

  // when tooltip is clicked
  function onTooltipClick(event) {
    // when user clicks on tooltip, stop click from transferring
    // outside of tooltip (eg, click off to close tooltip, or eg click
    // off to unhighlight same refs)
    event.stopPropagation();
  }

  // when tooltip is unhovered
  function onTooltipUnhover(event) {
    if (clickClose === "true") return;

    // make sure new mouse/touch/focus no longer over tooltip or any
    // element within it
    const tooltip = document.getElementById("tooltip");
    if (!tooltip) return;
    if (this.contains(event.relatedTarget)) return;

    closeTooltip();
  }

  // make nav bar to go betwen prev/next instances of same reference
  function makeNavBar(link) {
    // find other links to the same source
    const sameLinks = getSameLinks(link);

    // don't show nav bar when singular reference
    if (sameLinks.total <= 1) return;

    // find prev/next links with same target
    const prevLink = getPrevLink(link, sameLinks);
    const nextLink = getNextLink(link, sameLinks);

    // create nav bar
    const navBar = document.createElement("div");
    navBar.id = "tooltip_nav_bar";
    const text = sameLinks.index + 1 + " of " + sameLinks.total;

    // create nav bar prev/next buttons
    const prevButton = document.createElement("button");
    const nextButton = document.createElement("button");
    prevButton.id = "tooltip_prev_button";
    nextButton.id = "tooltip_next_button";
    prevButton.title =
      "Jump to the previous occurence of this item in the document [←]";
    nextButton.title =
      "Jump to the next occurence of this item in the document [→]";
    prevButton.classList.add("icon_button");
    nextButton.classList.add("icon_button");
    prevButton.innerHTML = document.querySelector(".icon_caret_left").innerHTML;
    nextButton.innerHTML =
      document.querySelector(".icon_caret_right").innerHTML;
    navBar.appendChild(prevButton);
    navBar.appendChild(document.createTextNode(text));
    navBar.appendChild(nextButton);

    // attach click listeners to buttons
    prevButton.addEventListener("click", function () {
      onPrevNextClick(link, prevLink);
    });
    nextButton.addEventListener("click", function () {
      onPrevNextClick(link, nextLink);
    });

    return navBar;
  }

  // get previous link with same target
  function getPrevLink(link, sameLinks) {
    if (!sameLinks) sameLinks = getSameLinks(link);
    // wrap index to other side if < 1
    let index;
    if (sameLinks.index - 1 >= 0) index = sameLinks.index - 1;
    else index = sameLinks.total - 1;
    return sameLinks.elements[index];
  }

  // get next link with same target
  function getNextLink(link, sameLinks) {
    if (!sameLinks) sameLinks = getSameLinks(link);
    // wrap index to other side if > total
    let index;
    if (sameLinks.index + 1 <= sameLinks.total - 1) index = sameLinks.index + 1;
    else index = 0;
    return sameLinks.elements[index];
  }

  // get element that is target of link or url hash
  function getSource(link) {
    const hash = link ? link.hash : window.location.hash;
    const id = hash.slice(1);
    let target = document.querySelector('[id="' + id + '"]');
    if (!target) return;

    // if ref or figure, modify target to get expected element
    if (id.indexOf("ref-") === 0) target = target.querySelector(":nth-child(2)");
    else if (id.indexOf("fig:") === 0) target = target.querySelector("figure");

    return target;
  }

  // when prev/next arrow button is clicked
  function onPrevNextClick(link, prevNextLink) {
    if (link && prevNextLink)
      goToElement(prevNextLink, window.innerHeight * 0.5);
  }

  // scroll to and focus element
  function goToElement(element, offset) {
    // expand accordion section if collapsed
    expandElement(element);
    const y =
      getRectInView(element).top -
      getRectInView(document.documentElement).top -
      (offset || 0);
    // trigger any function listening for "onscroll" event
    window.dispatchEvent(new Event("scroll"));
    window.scrollTo(0, y);
    document.activeElement.blur();
    element.focus();
  }

  // determine position to place tooltip based on link position in
  // viewport and tooltip size
  function positionTooltip(link, left, top) {
    const tooltipElement = document.getElementById("tooltip");
    if (!tooltipElement) return;

    // get convenient vars for position/dimensions of
    // link/tooltip/page/view
    link = getRectInPage(link);
    const tooltip = getRectInPage(tooltipElement);
    const view = getRectInPage();

    // horizontal positioning
    if (left)
      // use explicit value
      left = left;
    else if (link.left + tooltip.width < view.right)
      // fit tooltip to right of link
      left = link.left;
    else if (link.right - tooltip.width > view.left)
      // fit tooltip to left of link
      left = link.right - tooltip.width;
    // center tooltip in view
    else left = (view.right - view.left) / 2 - tooltip.width / 2;

    // vertical positioning
    if (top)
      // use explicit value
      top = top;
    else if (link.top - tooltip.height > view.top)
      // fit tooltip above link
      top = link.top - tooltip.height;
    else if (link.bottom + tooltip.height < view.bottom)
      // fit tooltip below link
      top = link.bottom;
    else {
      // center tooltip in view
      top = view.top + view.height / 2 - tooltip.height / 2;
      // nudge off of link to left/right if possible
      if (link.right + tooltip.width < view.right) left = link.right;
      else if (link.left - tooltip.width > view.left)
        left = link.left - tooltip.width;
    }

    tooltipElement.style.left = left + "px";
    tooltipElement.style.top = top + "px";
  }

  // start script when document is finished loading
  window.addEventListener("load", start);
</script>

<!-- caret left icon -->

<template class="icon_caret_left">
  <!-- modified from: https://fontawesome.com/icons/caret-left -->
  <svg width="16" height="16" viewBox="0 0 192 512">
    <path
      fill="currentColor"
      d="M192 127.338v257.324c0 17.818-21.543 26.741-34.142 14.142L29.196 270.142c-7.81-7.81-7.81-20.474 0-28.284l128.662-128.662c12.599-12.6 34.142-3.676 34.142 14.142z"
    ></path>
  </svg>
</template>

<!-- caret right icon -->

<template class="icon_caret_right">
  <!-- modified from: https://fontawesome.com/icons/caret-right -->
  <svg width="16" height="16" viewBox="0 0 192 512">
    <path
      fill="currentColor"
      d="M0 384.662V127.338c0-17.818 21.543-26.741 34.142-14.142l128.662 128.662c7.81 7.81 7.81 20.474 0 28.284L34.142 398.804C21.543 411.404 0 402.48 0 384.662z"
    ></path>
  </svg>
</template>

<style>
  @media only screen {
    /* tooltip container */
    #tooltip {
      position: absolute;
      width: 50%;
      min-width: 240px;
      max-width: 75%;
      z-index: 1;
    }

    /* tooltip content */
    #tooltip_content {
      margin-bottom: 5px;
      padding: 20px;
      border-radius: 5px;
      border: solid 1px #bdbdbd;
      box-shadow: 0 0 20px rgba(0, 0, 0, 0.05);
      background: #ffffff;
      overflow-wrap: break-word;
    }

    /* tooltip copy of paragraphs and figures */
    #tooltip_content > p,
    #tooltip_content > figure {
      margin: 0;
      max-height: 320px;
      overflow-y: auto;
    }

    /* tooltip copy of <img> */
    #tooltip_content > figure > img,
    #tooltip_content > figure > svg {
      max-height: 260px;
    }

    /* navigation bar */
    #tooltip_nav_bar {
      margin-top: 10px;
      text-align: center;
    }

    /* navigation bar previous/next buton */
    #tooltip_nav_bar > .icon_button {
      position: relative;
      top: 3px;
    }

    /* navigation bar previous button */
    #tooltip_nav_bar > .icon_button:first-of-type {
      margin-right: 5px;
    }

    /* navigation bar next button */
    #tooltip_nav_bar > .icon_button:last-of-type {
      margin-left: 5px;
    }
  }

  /* always hide tooltip on print */
  @media only print {
    #tooltip {
      display: none;
    }
  }
</style>
<!--
  Analytics Plugin (third-party) 
  
  Copy and paste code from Google Analytics or similar service here.
-->
<!-- 
  Annotations Plugin

  Allows public annotation of the  manuscript. See https://web.hypothes.is/.
-->

<script type="module">
  // configuration
  window.hypothesisConfig = function () {
    return {
      branding: {
        accentColor: "#2196f3",
        appBackgroundColor: "#f8f8f8",
        ctaBackgroundColor: "#f8f8f8",
        ctaTextColor: "#000000",
        selectionFontFamily: "Open Sans, Helvetica, sans serif",
        annotationFontFamily: "Open Sans, Helvetica, sans serif",
      },
    };
  };

  // hypothesis client script
  const embed = "https://hypothes.is/embed.js";
  // hypothesis annotation count query url
  const query = "https://api.hypothes.is/api/search?limit=0&url=";

  // start script
  function start() {
    const button = makeButton();
    document.body.insertBefore(button, document.body.firstChild);
    insertCount(button);
  }

  // make button
  function makeButton() {
    // create button
    const button = document.createElement("button");
    button.id = "hypothesis_button";
    button.innerHTML = document.querySelector(".icon_hypothesis").innerHTML;
    button.title = "Hypothesis annotations";
    button.classList.add("icon_button");

    function onClick(event) {
      onButtonClick(event, button);
    }

    // attach click listeners
    button.addEventListener("click", onClick);

    return button;
  }

  // insert annotations count
  async function insertCount(button) {
    // get annotation count from Hypothesis based on url
    let count = "-";
    try {
      const canonical = document.querySelector('link[rel="canonical"]');
      const location = window.location;
      const url = encodeURIComponent((canonical || location).href);
      const response = await fetch(query + url);
      const json = await response.json();
      count = json.total || "-";
    } catch (error) {
      console.log(error);
    }

    // put count into button
    const counter = document.createElement("span");
    counter.id = "hypothesis_count";
    counter.innerHTML = count;
    button.title = "View " + count + " Hypothesis annotations";
    button.append(counter);
  }

  // when button is clicked
  function onButtonClick(event, button) {
    const script = document.createElement("script");
    script.src = embed;
    document.body.append(script);
    button.remove();
  }

  // start script when document is finished loading
  window.addEventListener("load", start);
</script>

<!-- hypothesis icon -->

<template class="icon_hypothesis">
  <!-- modified from: https://simpleicons.org/icons/hypothesis.svg / https://git.io/Jf1VB -->
  <svg width="16" height="16" viewBox="0 0 24 24" tabindex="-1">
    <path
      fill="currentColor"
      d="M3.43 0C2.5 0 1.72 .768 1.72 1.72V18.86C1.72 19.8 2.5 20.57 3.43 20.57H9.38L12 24L14.62 20.57H20.57C21.5 20.57 22.29 19.8 22.29 18.86V1.72C22.29 .77 21.5 0 20.57 0H3.43M5.14 3.43H7.72V9.43S8.58 7.72 10.28 7.72C12 7.72 13.74 8.57 13.74 11.24V17.14H11.16V12C11.16 10.61 10.28 10.07 9.43 10.29C8.57 10.5 7.72 11.41 7.72 13.29V17.14H5.14V3.43M18 13.72C18.95 13.72 19.72 14.5 19.72 15.42A1.71 1.71 0 0 1 18 17.13A1.71 1.71 0 0 1 16.29 15.42C16.29 14.5 17.05 13.71 18 13.71Z"
      tabindex="-1"
    ></path>
  </svg>
</template>

<style>
  /* hypothesis activation button */
  #hypothesis_button {
    box-sizing: border-box;
    position: fixed;
    top: 0;
    right: 0;
    width: 60px;
    height: 60px;
    background: #ffffff;
    border-radius: 0;
    border-left: solid 1px #bdbdbd;
    border-bottom: solid 1px #bdbdbd;
    box-shadow: 0 0 20px rgba(0, 0, 0, 0.05);
    z-index: 2;
  }

  /* hypothesis button svg */
  #hypothesis_button > svg {
    position: relative;
    top: -4px;
  }

  /* hypothesis annotation count */
  #hypothesis_count {
    position: absolute;
    left: 0;
    right: 0;
    bottom: 5px;
  }

  /* side panel */
  .annotator-frame {
    width: 280px !important;
  }

  /* match highlight color to rest of theme */
  .annotator-highlights-always-on .annotator-hl {
    background-color: #ffeb3b !important;
  }

  /* match focused color to rest of theme */
  .annotator-hl.annotator-hl-focused {
    background-color: #ff8a65 !important;
  }

  /* match bucket bar color to rest of theme */
  .annotator-bucket-bar {
    background: #f5f5f5 !important;
  }

  /* always hide button, toolbar, and tooltip on print */
  @media only print {
    #hypothesis_button {
      display: none;
    }

    .annotator-frame {
      display: none !important;
    }

    hypothesis-adder {
      display: none !important;
    }
  }
</style>
<!-- 
  Mathjax Plugin (third-party) 

  Allows the proper rendering of math/equations written in LaTeX.
  See https://www.mathjax.org/.
-->

<script type="text/x-mathjax-config">
  // configuration
  MathJax.Hub.Config({
    "CommonHTML": { linebreaks: { automatic: true } },
    "HTML-CSS": { linebreaks: { automatic: true } },
    "SVG": { linebreaks: { automatic: true } },
    "fast-preview": { disabled: true }
  });
</script>

<script
  src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js?config=TeX-MML-AM_CHTML"
  integrity="sha512-0doc9hKxR3PYwso42RD1p5ySZpzzuDiOwMrdCEh2WdJZCjcmFKc/wEnL+z8fBQrnHoiNWbo+3fiGkOYXBdQp4A=="
  crossorigin="anonymous"
></script>

<style>
  /* mathjax containers */
  .math.display > span:not(.MathJax_Preview) {
    /* turn inline element (no dimensions) into block (allows fixed width and thus scrolling) */
    display: flex !important;
    overflow-x: auto !important;
    overflow-y: hidden !important;
    justify-content: center;
    align-items: center;
    margin: 0 !important;
  }

  /* right click menu */
  .MathJax_Menu {
    border-radius: 5px !important;
    border: solid 1px #bdbdbd !important;
    box-shadow: none !important;
  }

  /* equation auto-number */
  span[id^="eq:"] > span.math.display + span {
    font-weight: 600;
  }

  /* equation */
  span[id^="eq:"] > span.math.display > span {
    /* nudge to make room for equation auto-number and anchor */
    margin-right: 60px !important;
  }
</style>
</body>
</html>
